



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="病原微生物高通量测序数据分析手册">
      
      
        <link rel="canonical" href="https://indexofire.github.io/pathongs/chapter_10/01_trace-Pae-from-different-source-by-SNP/">
      
      
        <meta name="author" content="Mark Renton">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-0.17.3, mkdocs-material-2.7.3">
    
    
      
        <title>10.1 铜绿假单胞菌不同来源株SNP溯源 - Pathogen NGS 数据分析入门</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application.8d40d89b.css">
      
    
    
      <script src="../../assets/javascripts/modernizr.1aa3b519.js"></script>
    
    
      <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
      
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <a href="#snp" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://indexofire.github.io/pathongs" title="Pathogen NGS 数据分析入门" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                Pathogen NGS 数据分析入门
              </span>
              <span class="md-header-nav__topic">
                10.1 铜绿假单胞菌不同来源株SNP溯源
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://indexofire.github.io/pathongs/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      Indexofire
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    <span class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </span>
    Pathogen NGS 数据分析入门
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://indexofire.github.io/pathongs/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      Indexofire
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="前言" class="md-nav__link">
      前言
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2">
    
    <label class="md-nav__link" for="nav-2">
      第一章 背景知识介绍
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        第一章 背景知识介绍
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_01/01_basic-cml/" title="01. Linux 系统操作基本命令" class="md-nav__link">
      01. Linux 系统操作基本命令
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_01/02_linux-installation/" title="02. Linux 安装实例" class="md-nav__link">
      02. Linux 安装实例
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_01/03_linux-administration/" title="03. Linux 系统管理" class="md-nav__link">
      03. Linux 系统管理
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_01/04_awk/" title="04. 数据处理常用工具之 awk" class="md-nav__link">
      04. 数据处理常用工具之 awk
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_01/05_bioawk/" title="05. 测序数据工具 bioawk" class="md-nav__link">
      05. 测序数据工具 bioawk
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_01/06_python/" title="06. 了解 Python 生活轻松一点" class="md-nav__link">
      06. 了解 Python 生活轻松一点
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_01/07_shell/" title="07. 生信分析要学一点 shell" class="md-nav__link">
      07. 生信分析要学一点 shell
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      第二章 序列处理
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        第二章 序列处理
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_02/01_emboss/" title="01. EMBOSS 工具集" class="md-nav__link">
      01. EMBOSS 工具集
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_02/02_blast/" title="02. Blast 与 Blast+" class="md-nav__link">
      02. Blast 与 Blast+
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_02/08_seqkit/" title="03. 国产序列处理神器 SeqKit" class="md-nav__link">
      03. 国产序列处理神器 SeqKit
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_02/09_kmer/" title="04. K-mer 处理工具" class="md-nav__link">
      04. K-mer 处理工具
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      第三章 数据的上传和下载
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        第三章 数据的上传和下载
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_03/01_databases/" title="01 生物公共数据库" class="md-nav__link">
      01 生物公共数据库
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_03/02_fileformat/" title="02 生物数据文件格式" class="md-nav__link">
      02 生物数据文件格式
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_03/03_edirect/" title="03. 玩转 edirect" class="md-nav__link">
      03. 玩转 edirect
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      第四章 基因组组装
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        第四章 基因组组装
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_04/03_bacteria-assembly/" title="03. 细菌基因组组装" class="md-nav__link">
      03. 细菌基因组组装
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_04/04_virus-assembly/" title="04. 病毒基因组组装" class="md-nav__link">
      04. 病毒基因组组装
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6">
    
    <label class="md-nav__link" for="nav-6">
      第五章 基因组序列注释
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        第五章 基因组序列注释
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_05/01_prokka/" title="01. Prokka 快速注释" class="md-nav__link">
      01. Prokka 快速注释
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_05/02_rast/" title="02. rast" class="md-nav__link">
      02. rast
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_05/03_PGAAP/" title="03. PGAAP" class="md-nav__link">
      03. PGAAP
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7" checked>
    
    <label class="md-nav__link" for="nav-7">
      第十章 基因组分析实践
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-7">
        第十章 基因组分析实践
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        10.1 铜绿假单胞菌不同来源株SNP溯源
      </label>
    
    <a href="./" title="10.1 铜绿假单胞菌不同来源株SNP溯源" class="md-nav__link md-nav__link--active">
      10.1 铜绿假单胞菌不同来源株SNP溯源
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1" title="1. 提出问题" class="md-nav__link">
    1. 提出问题
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2" title="2. 分析流程" class="md-nav__link">
    2. 分析流程
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#21" title="2.1 下载数据" class="md-nav__link">
    2.1 下载数据
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#22-st17" title="2.2 建立 ST17 型参考菌株索引" class="md-nav__link">
    2.2 建立 ST17 型参考菌株索引
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#23-reads-st17" title="2.3 将测序 reads 比对到 ST17 并排序索引" class="md-nav__link">
    2.3 将测序 reads 比对到 ST17 并排序索引
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#24-reads" title="2.4 绘制测序 reads 的覆盖度" class="md-nav__link">
    2.4 绘制测序 reads 的覆盖度
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#25-1snps" title="2.5 比较1号临床株与环境株的SNPs差异" class="md-nav__link">
    2.5 比较1号临床株与环境株的SNPs差异
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#26-2" title="2.6 2个临床株的耐药变迁" class="md-nav__link">
    2.6 2个临床株的耐药变迁
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-snippy" title="3. 用 Snippy 来实现" class="md-nav__link">
    3. 用 Snippy 来实现
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#31-snippy" title="3.1 安装 Snippy" class="md-nav__link">
    3.1 安装 Snippy
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#32-snippy" title="3.2 使用 Snippy" class="md-nav__link">
    3.2 使用 Snippy
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-workflow" title="4. 用 Workflow 脚本来简化操作" class="md-nav__link">
    4. 用 Workflow 脚本来简化操作
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#41" title="4.1 准备数据" class="md-nav__link">
    4.1 准备数据
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5-reference" title="5. 如何选择 Reference" class="md-nav__link">
    5. 如何选择 Reference
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6" title="6. 参考资料" class="md-nav__link">
    6. 参考资料
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../02_comparative-analysis-Ban-genomes/" title="10.2 炭疽芽胞杆菌比较基因组学研究" class="md-nav__link">
      10.2 炭疽芽胞杆菌比较基因组学研究
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1" title="1. 提出问题" class="md-nav__link">
    1. 提出问题
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2" title="2. 分析流程" class="md-nav__link">
    2. 分析流程
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#21" title="2.1 下载数据" class="md-nav__link">
    2.1 下载数据
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#22-st17" title="2.2 建立 ST17 型参考菌株索引" class="md-nav__link">
    2.2 建立 ST17 型参考菌株索引
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#23-reads-st17" title="2.3 将测序 reads 比对到 ST17 并排序索引" class="md-nav__link">
    2.3 将测序 reads 比对到 ST17 并排序索引
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#24-reads" title="2.4 绘制测序 reads 的覆盖度" class="md-nav__link">
    2.4 绘制测序 reads 的覆盖度
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#25-1snps" title="2.5 比较1号临床株与环境株的SNPs差异" class="md-nav__link">
    2.5 比较1号临床株与环境株的SNPs差异
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#26-2" title="2.6 2个临床株的耐药变迁" class="md-nav__link">
    2.6 2个临床株的耐药变迁
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-snippy" title="3. 用 Snippy 来实现" class="md-nav__link">
    3. 用 Snippy 来实现
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#31-snippy" title="3.1 安装 Snippy" class="md-nav__link">
    3.1 安装 Snippy
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#32-snippy" title="3.2 使用 Snippy" class="md-nav__link">
    3.2 使用 Snippy
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-workflow" title="4. 用 Workflow 脚本来简化操作" class="md-nav__link">
    4. 用 Workflow 脚本来简化操作
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#41" title="4.1 准备数据" class="md-nav__link">
    4.1 准备数据
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5-reference" title="5. 如何选择 Reference" class="md-nav__link">
    5. 如何选择 Reference
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6" title="6. 参考资料" class="md-nav__link">
    6. 参考资料
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="snp">铜绿假单胞菌临床株与环境株SNP溯源<a class="headerlink" href="#snp" title="Permanent link">&para;</a></h1>
<div class="admonition note">
<p class="admonition-title">本节实践</p>
<p>利用高通量测序技术对铜绿假单胞菌的临床分离株与环境分离株在基因组水平上进行SNP分析，从而溯源菌株并做耐药分析。</p>
</div>
<p><strong>分析方法</strong>：</p>
<p>利用 bwa 将目标基因组 reads 比对到参考基因组，用 freebayes 做 variant calling，获得各个样本的 SNPs，根据结果做进一步分析。</p>
<p><strong>应用工具</strong>:</p>
<ul>
<li>bwa</li>
<li>samtools</li>
<li>freebayes</li>
<li>vcffilter</li>
</ul>
<p><strong>分析流程</strong>:</p>
<p><strong>实践意义</strong>：</p>
<p>可以作为今后实验室开展病原菌溯源工作的一个技术指导依据。</p>
<h2 id="1">1. 提出问题<a class="headerlink" href="#1" title="Permanent link">&para;</a></h2>
<p>工作流程将从以下几个方面对于这个案例开展数据分析：</p>
<ul>
<li>测序质量：reads的长度，插入片段的长度</li>
<li>将测序数据比对到参考基因组ST17：</li>
<li>每个样品reads的分布情况</li>
<li>每个样品的基因组测序覆盖度</li>
<li>不同样品之间的差异，特别是当设置不同的过滤参数时的情况：<ul>
<li>Quality filter: QUAL &gt; 100</li>
<li>Likelihood filter: QUAL / AO &gt; 10</li>
<li>Variant frequency filter: AO / DP &gt; 0.9 (90% allele frequency)</li>
</ul>
</li>
<li>哪一个水龙头水样的分离株最有可能是感染源。(方法: 用<code class="codehilite">vcfsamplediff</code>比较环境株与临床株SNPs数量)</li>
<li>2个病人分离株之间使用了大量的抗生素治疗，哪一个获得、得了耐药性？</li>
<li>用 joint calling 来比较2个临床样品数据，并用<code class="codehilite">vcfsamplediff</code>分析</li>
<li>对于获得的大量SNPs，是否需要过滤以及如何过滤</li>
<li>了解SNPs的分布规律</li>
<li>重点查看SNPs</li>
</ul>
<h2 id="2">2. 分析流程<a class="headerlink" href="#2" title="Permanent link">&para;</a></h2>
<h3 id="21">2.1 下载数据<a class="headerlink" href="#21" title="Permanent link">&para;</a></h3>
<p>数据可以从<a href="http://www.microbesng.uk/filedist/pseudomonas-practical/">这里</a>下载获得，可以用<code class="codehilite">wget -m</code>递归下载路径下所有文件。文件包括：</p>
<ul>
<li>TAP1 - 病房1号水龙头水样中分离的菌株</li>
<li>TAP2 - 病房2号水龙头水样中分离的菌株</li>
<li>TAP3 - 病房3号水龙头水样中分离的菌株</li>
<li>PATIENT1 - 服用抗生素后的病人分离株</li>
<li>PATIENT2 - 一周内进一步服用抗生素的病人分离株</li>
</ul>
<div class="codehilite"><pre><span></span><span class="c1"># 新建一个目录，数据和操作均在里面进行</span>
$ mkdir pae-case <span class="o">&amp;&amp;</span> <span class="nb">cd</span> pae-case
<span class="c1"># 下载数据文件</span>
$ wget -m http://www.microbesng.uk/filedist/pseudomonas-practical/
$ mv www.microbesng.uk/filedist/pseudomonas-practical/* .
</pre></div>

<h3 id="22-st17">2.2 建立 ST17 型参考菌株索引<a class="headerlink" href="#22-st17" title="Permanent link">&para;</a></h3>
<p>软件使用<code class="codehilite">bwa</code>来对参考基因组ST17建立索引，以便后续比对。</p>
<div class="codehilite"><pre><span></span>$ bwa index ST17.fasta
</pre></div>

<h3 id="23-reads-st17">2.3 将测序 reads 比对到 ST17 并排序索引<a class="headerlink" href="#23-reads-st17" title="Permanent link">&para;</a></h3>
<p>用<code class="codehilite">bwa mem</code>进行比对，输出的结果通过管道由samtools转换成BAM文件并进行排序和索引</p>
<div class="codehilite"><pre><span></span>$ bwa mem -t <span class="m">4</span> -R <span class="s1">&#39;@RG\tID:TAP1\tSM:TAP1&#39;</span> ST17.fasta <span class="se">\</span>
&gt; TAP1_R1_paired.fastq.gz TAP1_R2_paired.fastq.gz <span class="p">|</span> <span class="se">\</span>
&gt; samtools view -bS - <span class="p">|</span> samtools sort - TAP1.sorted
$ samtools index TAP1.sorted.bam
</pre></div>

<div class="admonition note">
<p class="admonition-title">参数介绍</p>
<ul>
<li>-t 4: 表示使用4线程，根据自己的计算机CPU资源设置。</li>
<li>-R '...': 设置reads的ID，\t为制表符会转义成tab键。</li>
</ul>
</div>
<p>对其他2株环境株和2株临床株重复以上步骤。</p>
<h3 id="24-reads">2.4 绘制测序 reads 的覆盖度<a class="headerlink" href="#24-reads" title="Permanent link">&para;</a></h3>
<p>bam文件含有比对的信息，可以用<code class="codehilite">stats</code>参数查看</p>
<div class="codehilite"><pre><span></span>$ samtools stats TAP1.sorted.bam <span class="p">|</span> grep <span class="s2">&quot;maximum length&quot;</span>
$ samtools stats TAP1.sorted.bam <span class="p">|</span> grep <span class="s2">&quot;insert size&quot;</span>
$ samtools stats TAP1.sorted.bam <span class="p">|</span> grep <span class="s2">&quot;^COV&quot;</span> &gt; TAP1.coverage.txt
</pre></div>

<p>将截取的覆盖度相关信息保存成<code class="codehilite"><span class="na">.txt</span></code>文件，在R里用ggplot2绘制覆盖度图：</p>
<div class="codehilite"><pre><span></span><span class="o">&gt;</span> <span class="kn">library</span><span class="p">(</span>ggplot2<span class="p">)</span>
<span class="o">&gt;</span> cov<span class="o">=</span>read.table<span class="p">(</span><span class="s">&quot;TAP1.coverage.txt&quot;</span><span class="p">,</span> sep<span class="o">=</span><span class="s">&quot;\t&quot;</span><span class="p">)</span>
<span class="o">&gt;</span> cov<span class="p">[</span><span class="m">1</span><span class="p">,]</span>
<span class="o">&gt;</span> ggplot<span class="p">(</span>cov<span class="p">,</span> aes<span class="p">(</span>x<span class="o">=</span>V3<span class="p">,</span> y<span class="o">=</span>V4<span class="p">))</span> <span class="o">+</span> geom_bar<span class="p">(</span>stat<span class="o">=</span><span class="s">&quot;identity&quot;</span><span class="p">)</span> <span class="o">+</span> xlab<span class="p">(</span><span class="s">&quot;Coverage&quot;</span><span class="p">)</span> <span class="o">+</span> ylab<span class="p">(</span><span class="s">&quot;Count&quot;</span><span class="p">)</span>
</pre></div>

<p><img alt="coverage plot" src="../../assets/images/10.1/Rplot.png" /></p>
<h3 id="25-1snps">2.5 比较1号临床株与环境株的SNPs差异<a class="headerlink" href="#25-1snps" title="Permanent link">&para;</a></h3>
<p>用同样的方法对其他几个样本进行操作，获得sorted的bam格式文件，用<code class="codehilite">freebayes</code>对临床与环境株比较获得vcf文件。然后用vcflib工具过滤<code class="codehilite">QUAL/AO&gt;10</code>的部分(VCF元数据部分可以参考<a href="http://samtools.github.io/hts-specs/VCFv4.1.pdf">文档</a>)，并用<code class="codehilite">wc-l</code>计算行数统计结果。</p>
<div class="codehilite"><pre><span></span>$ freebayes -p <span class="m">1</span> -C <span class="m">5</span> -f ST17.fasta TAP1.sorted.bam PATIENT1.sorted.bam  &gt; compare_tap1.vcf
$ vcffilter -f <span class="s2">&quot;QUAL / AO &gt; 10&quot;</span> compare_tap1.vcf <span class="p">|</span> vcffilter -f <span class="s2">&quot;NS = 2&quot;</span> <span class="p">|</span> wc -l

<span class="m">6685</span>
</pre></div>

<p>获得临床株与环境株之间的SNP差异数量，结果表明这2株之间只有5个SNPs。</p>
<div class="codehilite"><pre><span></span>$ vcfsamplediff SAME TAP1 PATIENT1 compare_tap1.vcf <span class="p">|</span> vcffilter -f <span class="s2">&quot;QUAL / AO &gt; 10&quot;</span> <span class="p">|</span> vcffilter -f <span class="s2">&quot;NS = 2&quot;</span> <span class="p">|</span> vcffilter -f <span class="s2">&quot;! ( SAME = germline ) &quot;</span> <span class="p">|</span> grep -v <span class="s2">&quot;^#&quot;</span> <span class="p">|</span> wc -l

<span class="m">5</span>
</pre></div>

<p>用<code class="codehilite">vcfsamplediff</code>了解SNPs差异的基因，可以知道是那些gene/CDS里面发生了变化，还可以根据定位看突变是否是同义突变。如果snps数量接近无法区分，但是定位的基因有差异，那么菌株的溯源还需要进一步考虑。</p>
<div class="codehilite"><pre><span></span>$ vcfsamplediff SAME PATIENT1 TAP1 compare_tap1.vcf  <span class="p">|</span> vcffilter -f <span class="s2">&quot;QUAL / AO &gt; 10&quot;</span> <span class="p">|</span> vcffilter -f <span class="s2">&quot;NS = 2&quot;</span> <span class="p">|</span> vcffilter -f <span class="s2">&quot;! ( SAME = germline ) &quot;</span> &gt; tap_differences.vcf
$ bedtools intersect -a ST17.gff -b tap_differences.vcf <span class="p">|</span> awk -F<span class="s2">&quot;;&quot;</span> <span class="s1">&#39;{print NR, $2}&#39;</span> <span class="nv">OFS</span><span class="o">=</span><span class="s2">&quot;\t&quot;</span>
</pre></div>

<p>结果输出：</p>
<div class="codehilite"><pre><span></span><span class="m">1</span>   <span class="nv">gene</span><span class="o">=</span>yfiR
<span class="m">2</span>   <span class="nv">inference</span><span class="o">=</span>ab initio prediction:Prodigal:2.60,similar to AA sequence:RefSeq:YP_261793.1,protein motif:TIGRFAMs:TIGR03756,protein motif:Pfam:PF06834.5
<span class="m">3</span>   <span class="nv">inference</span><span class="o">=</span>ab initio prediction:Prodigal:2.60,similar to AA sequence:RefSeq:YP_261793.1,protein motif:TIGRFAMs:TIGR03756,protein motif:Pfam:PF06834.5
<span class="m">4</span>   <span class="nv">gene</span><span class="o">=</span>trbL
<span class="m">5</span>   <span class="nv">inference</span><span class="o">=</span>ab initio prediction:Prodigal:2.60,similar to AA sequence:RefSeq:YP_001349879.1,protein motif:Cdd:COG3481,protein motif:TIGRFAMs:TIGR03760,protein motif:Pfam:PF07514.5
</pre></div>

<h3 id="26-2">2.6 2个临床株的耐药变迁<a class="headerlink" href="#26-2" title="Permanent link">&para;</a></h3>
<p>了解2个临床株的SNPs差异，并查看这些SNPs所影响的gene</p>
<div class="codehilite"><pre><span></span>$ freebayes --ploidy <span class="m">1</span> -C <span class="m">5</span> -f ST17.fasta PATIENT1.sorted.bam PATIENT2.sorted.bam &gt; compare_patient.vcf
~$ vcfsamplediff SAME PATIENT1 TAP1 compare_patient.vcf  <span class="p">|</span> vcffilter -f <span class="s2">&quot;QUAL / AO &gt; 10&quot;</span> <span class="p">|</span> vcffilter -f <span class="s2">&quot;NS = 2&quot;</span> <span class="p">|</span> vcffilter -f <span class="s2">&quot;! ( SAME = germline ) &quot;</span> &gt; patient_differences.vcf
$ bedtools intersect -a ST17.gff -b patient_differences.vcf <span class="p">|</span> awk -F<span class="s2">&quot;;&quot;</span> <span class="s1">&#39;{print $2}&#39;</span>
</pre></div>

<hr />
<h2 id="3-snippy">3. 用 Snippy 来实现<a class="headerlink" href="#3-snippy" title="Permanent link">&para;</a></h2>
<p>大量的日常工作或者不熟悉各种命令行工具使用的话，就需要更简便的软件或者pipelines来帮助完成整个分析工作。因此这里我们使用<a href="https://github.com/tseemann/snippy/blob/master/bin/snippy">snippy</a>来实现分析SNPs，并绘制基于snps的进化树。</p>
<h3 id="31-snippy">3.1 安装 Snippy<a class="headerlink" href="#31-snippy" title="Permanent link">&para;</a></h3>
<p>Snippy是比较适合新手的工具，它提供了All in One的工具套装。虽然Snippy需要以下软件支持：</p>
<ul>
<li>Perl &gt;= 5.6</li>
<li>BioPerl &gt;= 1.6</li>
<li>bwa mem &gt;= 0.7.12</li>
<li>samtools &gt;= 1.1</li>
<li>GNU parallel &gt; 2013xxxx</li>
<li>freebayes &gt;= 0.9.20</li>
<li>freebayes sripts ( freebayes-parallel,fasta_generate_regions.py)</li>
<li>vcflib (vcffilter, vcfstreamsort, vcfuniq, vcffirstheader)</li>
<li>vcftools (vcf-consensus)</li>
<li>snpEff &gt;= 4.1</li>
</ul>
<p>但这些工具在Snippy安装包里都已经提供了，我们只需要根据自己的系统设置将以来工具添加到系统路径中去即可。对于perl的一些模块，可能需要更新后才能正常使用：<code class="codehilite">sudo cpan -u module_name</code></p>
<div class="codehilite"><pre><span></span>$ wget https://github.com/tseemann/snippy/archive/v2.9.tar.gz
$ tar xzf v2.9.tar.gz -C ~/app
$ <span class="nb">echo</span> <span class="s2">&quot;export PATH=</span><span class="nv">$PATH</span><span class="s2">:</span><span class="nv">$HOME</span><span class="s2">/app/snippy-2.9/bin:</span><span class="nv">$HOME</span><span class="s2">/app/snippy-2.9/binaries/linux/&quot;</span> &gt;&gt; /.bashrc
$ <span class="nb">source</span> ~/.bashrc
<span class="c1"># update all perl module</span>
$ sudo cpan -u
</pre></div>

<h3 id="32-snippy">3.2 使用 Snippy<a class="headerlink" href="#32-snippy" title="Permanent link">&para;</a></h3>
<p>Snippy 不仅可以获得SNP(包括MultiSNP)，也可以获得insertion, indeletion以及Comibination。文本格式的结果记录文件中的snps.tab中，也可以用浏览器打开snps.html查看。</p>
<div class="codehilite"><pre><span></span>$ snippy --cpus <span class="m">4</span> --outdir output1 --ref ST17.fasta <span class="se">\</span>
&gt; --R1 TAP1_R1_paired.fastq.gz --R2 TAP1_R2_paired.fastq.gz
$ head output/snps.tab

<span class="c1"># snps.txt对各种突变位点做一个数据统计</span>
$ cat output1/snps.txt
...
Software    snippy <span class="m">2</span>.9
Variant-COMPLEX <span class="m">1124</span>
Variant-DEL     <span class="m">27</span>
Variant-INS     <span class="m">43</span>
Variant-MNP     <span class="m">222</span>
Variant-SNP     <span class="m">3351</span>
VariantTotal    <span class="m">4767</span>

<span class="c1"># 计算不同</span>

<span class="c1"># 了解不同碱基的SNP变化数量，如T-&gt;C的SNP数量</span>
$ awk <span class="s1">&#39;$10==&quot;T=&gt;C&quot; {n++} END{print n}&#39;</span> output1/snps.gff
</pre></div>

<p>Snippy还可以生成多个基因组的共有SNPs的比对文件。用snippy分别生成3个TAP的snp列表数据到ouput*目录中，然后统计共有snps数量并生成snp序列文件aln。</p>
<div class="codehilite"><pre><span></span><span class="c1"># for old ubuntu system you need update outdated perl module List::Util</span>
$ sudo cpan -u List::Util
$ snippy-core --prefix core output1 output2 output3
...
Found <span class="m">30711</span> core SNPs from <span class="m">42075</span> variant sites.
Saved SNP table: core.tab
Constructing alignment object <span class="k">for</span> core.aln
...
</pre></div>

<p>如果要用raxml构建进化树，那获得的.aln文件还需要转换成.phy才可以使用。很多工具可以实现转换，这里使用网络服务来实现：<a href="http://sing.ei.uvigo.es/ALTER/">http://sing.ei.uvigo.es/ALTER/</a></p>
<div class="codehilite"><pre><span></span><span class="c1"># RAxML 构建进化树</span>
$ raxmlHPC -f a -x <span class="m">12345</span> -p <span class="m">12345</span> -# <span class="m">100</span> -m GTRGAMMA -s core.phy -n pae -T <span class="m">40</span>
$ figtree RAxML_bestTree.pae
</pre></div>

<p>生成的进化树图类似下图</p>
<p><img alt="Alt text" src="../RAxML_bestTree.ex.png" /></p>
<blockquote>
<p>因为比对的是fasta格式的文件，所以snps不止是编码区CDS的，而是整个基因组上的snps。</p>
</blockquote>
<h2 id="4-workflow">4. 用 Workflow 脚本来简化操作<a class="headerlink" href="#4-workflow" title="Permanent link">&para;</a></h2>
<p>对于我们实验室平时会有许多类似的SNP鉴定工作，那么下机数据要一个个操作会很繁琐，有时候我们通过 unix 的 pipeline 来简化或者写 pipeline 脚本来实现。而另一种解决方式是通过 Workflow 工具来自动化数据分析。专门的 Workflow 软件有许多，有些带GUI（如 galaxy 就可以建立图形化的 workflow），有些基于CML。对于在服务器上的生物信息学分析，我们一般都是丢给命令行来实现。Workflow 因此，这里使用基于 python 的 ruffus 来建立 workflow，作为一个示例演示如何简化平时大量重复的工作。</p>
<p>如果想了解更多关于 workflow/pipeline 的软件，可以查看以下网站：
 - <a href="https://github.com/common-workflow-language/common-workflow-language/wiki/Existing-Workflow-systems">common-workflow</a>
 - <a href="https://github.com/pditommaso/awesome-pipeline">awesome-pipeline</a></p>
<h3 id="41">4.1 准备数据<a class="headerlink" href="#41" title="Permanent link">&para;</a></h3>
<blockquote>
<p>undone</p>
</blockquote>
<h2 id="5-reference">5. 如何选择 Reference<a class="headerlink" href="#5-reference" title="Permanent link">&para;</a></h2>
<p>在做微生物SNP calling时，选择不同reference对结果是否会有影响？会有多大影响？这是具体分析时需要考虑的问题。有一些<a href="http://www.ncbi.nlm.nih.gov/pubmed/25144537">文章</a>进行了讨论。对于我们实验室开展病原微生物溯源时，由于菌株一般是高度近源的，所以影响不大。对于做进化关系分析的，还是需要考虑snp的假阳性问题，通常是要对snp做 filtering。而且对于不同物种考虑其他的突变事件与重组在进化上的影响。</p>
<p>另一方面不同软件的流程与参数不同，对于不熟悉生物信息学的生物学家当使用<code class="codehilite">All in One</code>之类的套件工具分析时，可能会发现不同软件的结果会有不小的差异。因此还是要从自己研究的物种特点角度去选择软件，所以说最好对于这类工具的具体细节需要进行了解，特别是软件的参数。那么对于文档不全或者不够丰富的软件来说，就要仔细研读它的源代码来了解其实现方式，这也就是学习一些shell、python、perl等语言的原因之一了。</p>
<h2 id="6">6. 参考资料<a class="headerlink" href="#6" title="Permanent link">&para;</a></h2>
<ol>
<li><a href="http://nbviewer.jupyter.org/github/nickloman/nickloman.github.com/blob/master/tutorials/Pseudomonas-practical.ipynb">Jupyter Notebook for Pseudomonas Practical</a></li>
<li><a href="http://bmjopen.bmj.com/content/4/11/e006278.full">Seeking the source of Pseudomonas aeruginosa infections in a recently opened hospital: an observational study using whole-genome sequencing.</a></li>
<li><a href="https://github.com/ekg/vcflib">Vcflib Git Repository</a></li>
<li><a href="https://github.com/ekg/freebayes">Freebayes Git Repository</a></li>
<li><a href="https://github.com/tseemann/snippy">Snippy</a></li>
<li><a href="http://samtools.github.io/hts-specs/VCFv4.1.pdf">VCF Document v4.1</a></li>
</ol>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../../chapter_05/03_PGAAP/" title="03. PGAAP" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                03. PGAAP
              </span>
            </div>
          </a>
        
        
          <a href="../02_comparative-analysis-Ban-genomes/" title="10.2 炭疽芽胞杆菌比较基因组学研究" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                10.2 炭疽芽胞杆菌比较基因组学研究
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2016 - 2018 HZCDC@Lab
          </div>
        
        powered by
        <a href="http://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application.b438e6c5.js"></script>
      
      <script>app.initialize({version:"0.17.3",url:{base:"../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/raphael/2.2.7/raphael.min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/underscore.js/1.8.3/underscore-min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/js-sequence-diagrams/1.0.6/sequence-diagram-min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/flowchart/1.6.5/flowchart.min.js"></script>
      
        <script src="../../assets/js/umlconvert.js"></script>
      
    
    
      
        <script>!function(e,a,t,n,o,c,i){e.GoogleAnalyticsObject=o,e.ga=e.ga||function(){(e.ga.q=e.ga.q||[]).push(arguments)},e.ga.l=1*new Date,c=a.createElement(t),i=a.getElementsByTagName(t)[0],c.async=1,c.src="https://www.google-analytics.com/analytics.js",i.parentNode.insertBefore(c,i)}(window,document,"script",0,"ga"),ga("create","UA-55420348-2","auto"),ga("set","anonymizeIp",!0),ga("send","pageview");var links=document.getElementsByTagName("a");if(Array.prototype.map.call(links,function(e){e.host!=document.location.host&&e.addEventListener("click",function(){var a=e.getAttribute("data-md-action")||"follow";ga("send","event","outbound",a,e.href)})}),document.forms.search){var query=document.forms.search.query;query.addEventListener("blur",function(){if(this.value){var e=document.location.pathname;ga("send","pageview",e+"?q="+this.value)}})}</script>
      
    
  </body>
</html>