



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="病原微生物高通量测序数据分析手册">
      
      
        <link rel="canonical" href="https://pathongs.hzcdclab.org/C16_Analysis-Examples/02_comparative-analysis-Ban-genomes/">
      
      
        <meta name="author" content="Mark Renton">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-0.17.1, mkdocs-material-2.7.3">
    
    
      
        <title>02. 细菌比较基因组学研究 - Pathogen NGS 数据分析入门</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application.8d40d89b.css">
      
    
    
      <script src="../../assets/javascripts/modernizr.1aa3b519.js"></script>
    
    
      <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
      
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <a href="#snps" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://pathongs.hzcdclab.org/" title="Pathogen NGS 数据分析入门" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                Pathogen NGS 数据分析入门
              </span>
              <span class="md-header-nav__topic">
                02. 细菌比较基因组学研究
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/indexofire/pathongs/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      GitHub
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    <span class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </span>
    Pathogen NGS 数据分析入门
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/indexofire/pathongs/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      GitHub
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="第零章 开始之前" class="md-nav__link">
      第零章 开始之前
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2">
    
    <label class="md-nav__link" for="nav-2">
      第一章 背景知识介绍
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        第一章 背景知识介绍
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/01_linux-installation/" title="01. Linux 安装实例" class="md-nav__link">
      01. Linux 安装实例
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/02_basic-cml/" title="02. Linux 基本命令" class="md-nav__link">
      02. Linux 基本命令
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/03_linux-software/" title="03. Linux 软件管理" class="md-nav__link">
      03. Linux 软件管理
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/04_awk/" title="04. awk 列处理工具" class="md-nav__link">
      04. awk 列处理工具
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/05_sed/" title="05. sed 行处理工具" class="md-nav__link">
      05. sed 行处理工具
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/06_tools/" title="06. 其他命令行工具" class="md-nav__link">
      06. 其他命令行工具
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/07_shell/" title="07. Shell脚本" class="md-nav__link">
      07. Shell脚本
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/08_perl/" title="08. Perl 编程" class="md-nav__link">
      08. Perl 编程
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/09_python/" title="09. Python 编程" class="md-nav__link">
      09. Python 编程
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/10_biopython/" title="10. Biopython 编程" class="md-nav__link">
      10. Biopython 编程
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/11_r/" title="11. R 编程" class="md-nav__link">
      11. R 编程
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      第二章 常用工具
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        第二章 常用工具
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C02_Common-Utility/01_databases/" title="01. 公共生物数据库" class="md-nav__link">
      01. 公共生物数据库
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C02_Common-Utility/02_blast/" title="02. Blast 与 NCBI_Blast" class="md-nav__link">
      02. Blast 与 NCBI_Blast
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C02_Common-Utility/03_edirect/" title="03. 玩转 edirect" class="md-nav__link">
      03. 玩转 edirect
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C02_Common-Utility/04_emboss/" title="04. EMBOSS 工具集" class="md-nav__link">
      04. EMBOSS 工具集
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C02_Common-Utility/07_mummer/" title="07. MUMmer 工具" class="md-nav__link">
      07. MUMmer 工具
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C02_Common-Utility/08_seqkit/" title="08. SeqKit 工具" class="md-nav__link">
      08. SeqKit 工具
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C02_Common-Utility/09_kmer/" title="09. Kmer 工具" class="md-nav__link">
      09. Kmer 工具
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      第三章 各类数据文件格式
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        第三章 各类数据文件格式
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C03_File-Format/01_fasta/" title="01. fasta" class="md-nav__link">
      01. fasta
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C03_File-Format/02_embl/" title="02. embl" class="md-nav__link">
      02. embl
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C03_File-Format/03_plain-txt/" title="03. plain text" class="md-nav__link">
      03. plain text
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      第四章 基因组组装
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        第四章 基因组组装
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C04_Pathogen-Genome-Assembly/01_bwa-samtools-pipeline/" title="01. bwa samtools 流程" class="md-nav__link">
      01. bwa samtools 流程
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C04_Pathogen-Genome-Assembly/02_gatk/" title="02. GATK 流程" class="md-nav__link">
      02. GATK 流程
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C04_Pathogen-Genome-Assembly/03_bacteria-assembly/" title="03. 细菌基因组组装" class="md-nav__link">
      03. 细菌基因组组装
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C04_Pathogen-Genome-Assembly/04_virus-assembly/" title="04. 病毒基因组组装" class="md-nav__link">
      04. 病毒基因组组装
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C04_Pathogen-Genome-Assembly/05_quast/" title="05. Quast 验证" class="md-nav__link">
      05. Quast 验证
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6">
    
    <label class="md-nav__link" for="nav-6">
      第五章 基因组比对
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        第五章 基因组比对
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C05_Pathogen-Mapping-Pipeline/01_bwa-long-sequence-mapping/" title="01. bwa 长序列比对" class="md-nav__link">
      01. bwa 长序列比对
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7">
    
    <label class="md-nav__link" for="nav-7">
      第六章 基因组注释
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-7">
        第六章 基因组注释
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C06_Genome-Annotation/01_prokka/" title="01. Prokka 快速注释" class="md-nav__link">
      01. Prokka 快速注释
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C06_Genome-Annotation/02_rast/" title="02. 在线注释工具 rast" class="md-nav__link">
      02. 在线注释工具 rast
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C06_Genome-Annotation/03_PGAAP/" title="03. 注释工具 PGAAP" class="md-nav__link">
      03. 注释工具 PGAAP
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C06_Genome-Annotation/04_snpEFF/" title="04. SNP注释工具 snpEFF" class="md-nav__link">
      04. SNP注释工具 snpEFF
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-8" type="checkbox" id="nav-8">
    
    <label class="md-nav__link" for="nav-8">
      第七章 泛基因组分析
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-8">
        第七章 泛基因组分析
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C07_PanGenome-and-CoreGenome/01_pangenome/" title="01. Pangenome 概述" class="md-nav__link">
      01. Pangenome 概述
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C07_PanGenome-and-CoreGenome/02_get_homologues/" title="02. Get_Homologues 分析" class="md-nav__link">
      02. Get_Homologues 分析
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C07_PanGenome-and-CoreGenome/03_BPGA/" title="03. BPGA 分析" class="md-nav__link">
      03. BPGA 分析
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C07_PanGenome-and-CoreGenome/04_roary/" title="04. Roary 分析" class="md-nav__link">
      04. Roary 分析
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C07_PanGenome-and-CoreGenome/05_panseq/" title="05. panseq 分析" class="md-nav__link">
      05. panseq 分析
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C07_PanGenome-and-CoreGenome/06_gubbins/" title="06. Gubbins 分析" class="md-nav__link">
      06. Gubbins 分析
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C07_PanGenome-and-CoreGenome/07_orthmcl/" title="07. OrthMCL 分析" class="md-nav__link">
      07. OrthMCL 分析
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C07_PanGenome-and-CoreGenome/08_ITEP/" title="08. ITEP 分析" class="md-nav__link">
      08. ITEP 分析
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-9" type="checkbox" id="nav-9">
    
    <label class="md-nav__link" for="nav-9">
      第八章 基因组可视化
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-9">
        第八章 基因组可视化
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C08_Genome-Visualization/01_igv/" title="01. igv 查看比对结果" class="md-nav__link">
      01. igv 查看比对结果
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C08_Genome-Visualization/02_BRIG/" title="02. BRIG 分析工具" class="md-nav__link">
      02. BRIG 分析工具
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C08_Genome-Visualization/03_circos/" title="03. circos 工具使用" class="md-nav__link">
      03. circos 工具使用
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-10" type="checkbox" id="nav-10">
    
    <label class="md-nav__link" for="nav-10">
      第九章 分析平台
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-10">
        第九章 分析平台
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C09_Analysis-Platform/01_docker/" title="01. 基于 docker 镜像" class="md-nav__link">
      01. 基于 docker 镜像
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C09_Analysis-Platform/02_galaxy/" title="02. Galaxy 平台" class="md-nav__link">
      02. Galaxy 平台
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C09_Analysis-Platform/03_amazon/" title="03. Amazon EC2 云平台" class="md-nav__link">
      03. Amazon EC2 云平台
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C09_Analysis-Platform/04_others/" title="04. 其他平台" class="md-nav__link">
      04. 其他平台
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-11" type="checkbox" id="nav-11">
    
    <label class="md-nav__link" for="nav-11">
      第十章 重组与种群分析
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-11">
        第十章 重组与种群分析
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C10_Recombination-and-Population-Structure/01_virus-recombination/" title="01. 病毒重组" class="md-nav__link">
      01. 病毒重组
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-12" type="checkbox" id="nav-12">
    
    <label class="md-nav__link" for="nav-12">
      第十一章 常见分析工具
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-12">
        第十一章 常见分析工具
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C11_Insilico-Genome-Application/01_srst2/" title="01. SRST2 短序列比对基因工具" class="md-nav__link">
      01. SRST2 短序列比对基因工具
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C11_Insilico-Genome-Application/02_mlst/" title="02. 拼接序列管家基因扫描工具 MLST" class="md-nav__link">
      02. 拼接序列管家基因扫描工具 MLST
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C11_Insilico-Genome-Application/03_ariba/" title="03. 拼接序列耐药基因扫描工具 Ariba" class="md-nav__link">
      03. 拼接序列耐药基因扫描工具 Ariba
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C11_Insilico-Genome-Application/04_csibelia/" title="04. 细菌共线性分析工具 C-Sibelia" class="md-nav__link">
      04. 细菌共线性分析工具 C-Sibelia
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-13" type="checkbox" id="nav-13">
    
    <label class="md-nav__link" for="nav-13">
      第十二章 宏基因组分析
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-13">
        第十二章 宏基因组分析
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/01_kraken/" title="01. kraken" class="md-nav__link">
      01. kraken
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/02_centrifuge/" title="02. centrifuge" class="md-nav__link">
      02. centrifuge
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/03_bracken/" title="03. bracken" class="md-nav__link">
      03. bracken
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/04_sourmash/" title="04. sourmash" class="md-nav__link">
      04. sourmash
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/05_clark/" title="05. clark" class="md-nav__link">
      05. clark
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/06_krona/" title="06. krona" class="md-nav__link">
      06. krona
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/07_pavian/" title="07. pavian" class="md-nav__link">
      07. pavian
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/08_supri/" title="08. supri" class="md-nav__link">
      08. supri
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/09_shotgun/" title="‘09. shotgun 分析’" class="md-nav__link">
      ‘09. shotgun 分析’
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/10_16s/" title="10. 16S 种群分析" class="md-nav__link">
      10. 16S 种群分析
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-14" type="checkbox" id="nav-14">
    
    <label class="md-nav__link" for="nav-14">
      第十三章 单分子测序
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-14">
        第十三章 单分子测序
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C13_Single-Molecular-Sequencing/01_nanopore/" title="01. nanopore" class="md-nav__link">
      01. nanopore
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C13_Single-Molecular-Sequencing/02_pacbio/" title="02. pacbio" class="md-nav__link">
      02. pacbio
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C13_Single-Molecular-Sequencing/03_canu/" title="03. canu" class="md-nav__link">
      03. canu
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C13_Single-Molecular-Sequencing/04_trimming/" title="04. trimming" class="md-nav__link">
      04. trimming
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C13_Single-Molecular-Sequencing/06_hybrid/" title="06. hybrid" class="md-nav__link">
      06. hybrid
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-15" type="checkbox" id="nav-15">
    
    <label class="md-nav__link" for="nav-15">
      第十四章 进化树构建
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-15">
        第十四章 进化树构建
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C14_Phylogeny-Analysis/01_msa/" title="01. 多重序列比对" class="md-nav__link">
      01. 多重序列比对
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C14_Phylogeny-Analysis/02_raxml/" title="02. RAxML" class="md-nav__link">
      02. RAxML
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-16" type="checkbox" id="nav-16">
    
    <label class="md-nav__link" for="nav-16">
      第十五章 基因组流行病学
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-16">
        第十五章 基因组流行病学
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C15_Pathogen-Geospatial-Analysis/04_cartodb/" title="04. CartoDB" class="md-nav__link">
      04. CartoDB
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-17" type="checkbox" id="nav-17" checked>
    
    <label class="md-nav__link" for="nav-17">
      第十六章 基因组分析实践
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-17">
        第十六章 基因组分析实践
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../01_trace-Pae-from-different-source-by-SNP/" title="01. 细菌SNP溯源分析一" class="md-nav__link">
      01. 细菌SNP溯源分析一
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        02. 细菌比较基因组学研究
      </label>
    
    <a href="./" title="02. 细菌比较基因组学研究" class="md-nav__link md-nav__link--active">
      02. 细菌比较基因组学研究
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1" title="1. 获取公共数据库数据" class="md-nav__link">
    1. 获取公共数据库数据
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#11-sra" title="1.1 下载 sra" class="md-nav__link">
    1.1 下载 sra
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#12-pese" title="1.2 区分 PE/SE 数据" class="md-nav__link">
    1.2 区分 PE/SE 数据
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#13-fastqgz" title="1.3 验证 fastq.gz 文件名的一致性" class="md-nav__link">
    1.3 验证 fastq.gz 文件名的一致性
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2" title="2. 数据的前期处理" class="md-nav__link">
    2. 数据的前期处理
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#21-qc" title="2.1 数据 QC" class="md-nav__link">
    2.1 数据 QC
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#22" title="2.2 去除接头" class="md-nav__link">
    2.2 去除接头
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-pangenome" title="3. Pangenome 分析" class="md-nav__link">
    3. Pangenome 分析
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#31" title="3.1 基因组组装" class="md-nav__link">
    3.1 基因组组装
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#32" title="3.2 基因组注释" class="md-nav__link">
    3.2 基因组注释
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#33-roary" title="3.3 用 Roary 进行分析" class="md-nav__link">
    3.3 用 Roary 进行分析
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#34-harvest" title="3.4 Harvest" class="md-nav__link">
    3.4 Harvest
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-snps" title="4. SNPs 构建进化树" class="md-nav__link">
    4. SNPs 构建进化树
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#41-mapping" title="4.1 基于参考基因组 Mapping 的方法" class="md-nav__link">
    4.1 基于参考基因组 Mapping 的方法
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-snippy" title="1. Snippy*" class="md-nav__link">
    1. Snippy*
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5" title="5. 核心基因分析" class="md-nav__link">
    5. 核心基因分析
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#51-get_homologues" title="5.1 get_homologues" class="md-nav__link">
    5.1 get_homologues
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6" title="6. 相关软件安装" class="md-nav__link">
    6. 相关软件安装
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#7-reference" title="7. Reference" class="md-nav__link">
    7. Reference
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1" title="1. 获取公共数据库数据" class="md-nav__link">
    1. 获取公共数据库数据
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#11-sra" title="1.1 下载 sra" class="md-nav__link">
    1.1 下载 sra
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#12-pese" title="1.2 区分 PE/SE 数据" class="md-nav__link">
    1.2 区分 PE/SE 数据
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#13-fastqgz" title="1.3 验证 fastq.gz 文件名的一致性" class="md-nav__link">
    1.3 验证 fastq.gz 文件名的一致性
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2" title="2. 数据的前期处理" class="md-nav__link">
    2. 数据的前期处理
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#21-qc" title="2.1 数据 QC" class="md-nav__link">
    2.1 数据 QC
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#22" title="2.2 去除接头" class="md-nav__link">
    2.2 去除接头
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-pangenome" title="3. Pangenome 分析" class="md-nav__link">
    3. Pangenome 分析
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#31" title="3.1 基因组组装" class="md-nav__link">
    3.1 基因组组装
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#32" title="3.2 基因组注释" class="md-nav__link">
    3.2 基因组注释
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#33-roary" title="3.3 用 Roary 进行分析" class="md-nav__link">
    3.3 用 Roary 进行分析
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#34-harvest" title="3.4 Harvest" class="md-nav__link">
    3.4 Harvest
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-snps" title="4. SNPs 构建进化树" class="md-nav__link">
    4. SNPs 构建进化树
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#41-mapping" title="4.1 基于参考基因组 Mapping 的方法" class="md-nav__link">
    4.1 基于参考基因组 Mapping 的方法
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-snippy" title="1. Snippy*" class="md-nav__link">
    1. Snippy*
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5" title="5. 核心基因分析" class="md-nav__link">
    5. 核心基因分析
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#51-get_homologues" title="5.1 get_homologues" class="md-nav__link">
    5.1 get_homologues
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#6" title="6. 相关软件安装" class="md-nav__link">
    6. 相关软件安装
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#7-reference" title="7. Reference" class="md-nav__link">
    7. Reference
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/indexofire/pathongs/edit/master/docs/C16_Analysis-Examples/02_comparative-analysis-Ban-genomes.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="snps">炭疽芽胞杆菌基因组 SNPs 分析<a class="headerlink" href="#snps" title="Permanent link">&para;</a></h1>
<div class="admonition note">
<p class="admonition-title">背景简介</p>
<p>做为疾控中心的实验室，对生物恐怖事件的防控和应对处置能力是应急工作能力的重要组成之一。而细菌性生物恐怖战剂中最著名的就是含有炭疽芽胞杆菌的白色粉末。做为基层实验室，因为在日常工作中接触到这些样品和病原的机会微乎其微，所以经验和技能方面较为缺乏。通过平时加强相关项目的应急演练外，实验室快速分析和溯源工作需要做好能力储备，因此实验室对炭疽芽胞杆菌的基因组数据进行了基本分析，对相关资料和信息进行了储备。</p>
</div>
<p>本节主要通过构建相应的分析流程，获得炭疽芽胞杆菌菌株基因组数据，可以第一时间就进行溯源方面的研究。将这个工作的过程建立文档保存。分析流程粗略简单，主要目的也是为实验室内部教学，希望能做到举一反三，给其他细菌性病原微生物基因组溯源分析研究提供帮助。</p>
<p>软件中使用的一些参数是根据自己的服务器设置的，具体运行时需要根据计算机配置进行相应的调整。</p>
<div class="admonition note">
<p class="admonition-title">服务器配置</p>
<ul>
<li>CPU: E3 2650v3 x2</li>
<li>Memory: ECC 16G x8</li>
<li>Harddisk: 7200转 1T x4 组 Raid 5 + 1 Hotspare</li>
<li>System: Ubuntu 16.04 LTS 64bit</li>
</ul>
</div>
<h2 id="1">1. 获取公共数据库数据<a class="headerlink" href="#1" title="Permanent link">&para;</a></h2>
<p>数据采用 NCBI SRA 数据库中的公共数据来作分析。首先检索 NCBI SRA 数据库中所有使用 illumina 平台对炭疽芽胞杆菌基因组进行 PE 测序的 大于 100M 的 NGS 数据，并在下载这些 sra 数据后转换成 fastq.gz 格式。使用的工具主要为 <code class="codehilite">edirect</code> 和 <code class="codehilite">sra-tools</code>。</p>
<h3 id="11-sra">1.1 下载 sra<a class="headerlink" href="#11-sra" title="Permanent link">&para;</a></h3>
<div class="codehilite"><pre><span></span><span class="c1"># 下载炭疽芽胞杆菌 SRA 双末端测序数据</span>
$ esearch -db sra -query <span class="s1">&#39;&quot;Bacillus anthracis&quot;[Organism] AND \</span>
<span class="s1">&gt; (Hiseq[ALL] OR Miseq[ALL]) AND &quot;strategy wgs&quot;[Properties] AND paired[ALL]&#39;</span> <span class="p">|</span> <span class="se">\</span>
&gt; efetch -db sra -format runinfo <span class="p">|</span> grep <span class="s1">&#39;^[DSE]RR*&#39;</span> <span class="p">|</span> <span class="se">\</span>
&gt; awk -F<span class="s1">&#39;,&#39;</span> <span class="s1">&#39;{if($5&gt;100000000) print $1}&#39;</span> <span class="p">|</span> sort -n <span class="p">|</span> <span class="se">\</span>
&gt; xargs fastq-dump --split-files --gzip --outdir bacillus_anthracis
</pre></div>

<p>上面的命令会直接调用 <code class="codehilite">fastq-dump</code> 下载数据到 <code class="codehilite">--outdir</code> 中。虽然我们用了 <code class="codehilite">grep PAIRED</code> 来区分不同的测序方式，但是由于信息是提交者上传的，可能信息不一定是正确的。由于后面的流程中牵涉到组装和比对等操作，PE/SE的方式会有区别，最好还是进行验证后再下载，或者将PE/SE的数据区分存放。</p>
<div class="codehilite"><pre><span></span><span class="c1"># 先生成数据记录</span>
$ esearch -db sra -query <span class="s1">&#39;&quot;Bacillus anthracis&quot;[Organism] AND \</span>
<span class="s1">&gt; (Hiseq[ALL] OR Miseq[ALL]) AND &quot;strategy wgs&quot;[Properties]&#39;</span> <span class="p">|</span> <span class="se">\</span>
&gt; efetch -db sra -format runinfo <span class="p">|</span> grep <span class="s1">&#39;PAIRED&#39;</span> <span class="p">|</span> grep <span class="s1">&#39;^[DSE]RR*&#39;</span> <span class="p">|</span> <span class="se">\</span>
&gt; awk -F<span class="s1">&#39;,&#39;</span> <span class="s1">&#39;{if($5&gt;100000000) print $1}&#39;</span> <span class="p">|</span> sort -n &gt; ban-sra.txt

<span class="c1"># 下载 sra</span>
$ cat ban-sra.txt <span class="p">|</span> xargs prefetch
</pre></div>

<p>这样会下载 sra 文件到默认目录(~/.ncbi/public/sra)，然后用下面的方法来区分下载的数据到底是PE还是SE测序。</p>
<h3 id="12-pese">1.2 区分 PE/SE 数据<a class="headerlink" href="#12-pese" title="Permanent link">&para;</a></h3>
<p>虽然我们检索时限定了 PE 测序数据，但是很多时候数据库里的信息不准确，还是会有很多 SE 的数据会被下载。确定 PE 或者 SE 的方法很多。</p>
<div class="codehilite"><pre><span></span><span class="c1"># sra-stat 统计 sra 文件，nreads=1 表示 SE 测序，nreads=2 表示 PE 测序。-e 2 只统计前2个spot，如果去掉-e参数，程序统计整个文件，速度比较慢</span>
$ <span class="k">for</span> i in *.sra<span class="p">;</span> <span class="k">do</span> <span class="nb">echo</span> <span class="nv">$i</span> <span class="sb">`</span>sra-stat -xs -e <span class="m">2</span> <span class="nv">$i</span> <span class="p">|</span> grep <span class="s2">&quot;nreads&quot;</span><span class="sb">`</span><span class="p">;</span> <span class="k">done</span>

<span class="c1"># 批量删除</span>
$ parallel <span class="s1">&#39;sra-stat -xs -e 2&#39;</span> ::: *.sra <span class="p">|</span> xtract -Pattern Run -element Run@accession <span class="se">\</span>
&gt; Statistics@nreads <span class="p">|</span> awk <span class="s1">&#39;{if($2==1) print $1}&#39;</span> <span class="p">|</span> xargs rm
</pre></div>

<p><code class="codehilite">sra-stat</code> 工具可以提取 sra 文件的测序相关信息。<code class="codehilite">-xs</code> 参数生成 xml 格式的统计信息。里面属性值 nreads 可以表示采用的是哪一种测序方式。如果 spot-id 是规则排列，那么可以添加 <code class="codehilite">-e 2</code> 最快时间获得结果。</p>
<p><code class="codehilite">parallel</code> 工具来并行化处理 <code class="codehilite">sra-stat</code>，如果计算量大是很好的利用多核CPU处理终端命令的方法。</p>
<div class="codehilite"><pre><span></span><span class="c1"># fastq-dump 输出一个 reads，计算行数如果是4，那么就是 SE，如果是8,则是 PE</span>
$ fastq-dump -X <span class="m">1</span> --split-spot -Z SRR955386.sra <span class="p">|</span> wc -l

<span class="c1"># 利用 -A 参数远程获得结果</span>
$ fastq-dump -X1 --split-spot -Z -L <span class="m">5</span> -A SRR955386

<span class="c1"># fastq-dump 自己判断，当 sra 文件是 SE 测序时，fastq-dump 只能生成1个 *_1.fastq 文件</span>
$ fastq-dump --split-files ERR493452.sra

<span class="c1"># fastq-dump 只能调用单核转换，可以使用 parallel 工具并行处理加快速度。</span>
$ parallel <span class="s2">&quot;fastq-dump --split-files --gzip --outdir bacillus_anthracis&quot;</span> ::: *.sra
</pre></div>

<p>另外一种获得是 PE/SE 的方法是用 <code class="codehilite">fastq-dump</code> 来读取出相同 spot-id 的 reads 再计算行数。<code class="codehilite">-X</code> 表示最多读取的 spot-id 数量；<code class="codehilite">--split-spot</code> 区分spot到reads然后用<code class="codehilite">-Z</code>输出，再<code class="codehilite">wc -l</code>统计行数。</p>
<p>有时候有些sra文件在 dump 时会出错，不能生成对应的 PE fastq 文件，这就需要比较后删除了。如果是 mate pair 的数据，会生成 <code class="codehilite">*_3.fastq.gz</code>，如果不需要则可手工删除。</p>
<h3 id="13-fastqgz">1.3 验证 fastq.gz 文件名的一致性<a class="headerlink" href="#13-fastqgz" title="Permanent link">&para;</a></h3>
<p><code class="codehilite">fastq-dump</code> 批量生成 fastq 数据后，有时因为各种原因，还是会 dump 出非 PE 类型的 <code class="codehilite">*._1</code> 和 <code class="codehilite">*._2</code> 命名风格的2个PE数据文件。文件量大时通过肉眼很难判断。</p>
<div class="codehilite"><pre><span></span><span class="c1"># 核对 PE 双端测序文件是否一致</span>
$ ls -l *_1.fastq.gz <span class="p">|</span> awk <span class="s1">&#39;{print $9}&#39;</span> <span class="p">|</span> awk -F<span class="s1">&#39;_&#39;</span> <span class="s1">&#39;{print $1}&#39;</span> &gt; R1.txt
$ ls -l *_2.fastq.gz <span class="p">|</span> awk <span class="s1">&#39;{print $9}&#39;</span> <span class="p">|</span> awk -F<span class="s1">&#39;_&#39;</span> <span class="s1">&#39;{print $1}&#39;</span> &gt; R2.txt
$ diff R1.txt R2.txt
</pre></div>

<p>通过 <code class="codehilite">diff</code> 工具比较，如果2个文件内容一致，则终端不会有输出。否则就要删除差异的那个文件了。</p>
<h2 id="2">2. 数据的前期处理<a class="headerlink" href="#2" title="Permanent link">&para;</a></h2>
<p>由于是公共数据库下载的数据，并不能保证测序实验质量或者数据提交者是否提交的是质控后的数据。因此建议要对数据做一些前期的质控处理。</p>
<h3 id="21-qc">2.1 数据 QC<a class="headerlink" href="#21-qc" title="Permanent link">&para;</a></h3>
<p>首先对基因组 GC 含量和 Q 值进行初步筛选，对于偏差较大的数据考虑直接剔除（或者用其他软件验证看是否是错误物种）。这里使用的工具为 <code class="codehilite">bioawk</code> 或 <code class="codehilite">parallel</code>。</p>
<div class="codehilite"><pre><span></span><span class="c1"># 因为 reads 覆盖度可能并不均一，高覆盖度的区域 GC 含量比重会略高。但是大部分情况下，正常测序的基因组这种区域相对来说不多，平均到基因组后会整体 GC 含量影响不大。</span>
$ <span class="k">for</span> i in *.fastq.gz<span class="p">;</span> <span class="k">do</span> bioawk -c fastx <span class="s1">&#39;BEGIN{n=0;q=0}{n+=gc($seq);q+=meanqual($seq)}END{print $name,n/NR,q/NR}&#39;</span> <span class="nv">$i</span> &gt;&gt; result.txt<span class="p">;</span> <span class="k">done</span>

<span class="c1"># awk 类工具是单进程的，为了加速可以使用 parallel 来并行计算</span>
$ parallel <span class="s2">&quot;bioawk -c fastx &#39;BEGIN{n=0;q=0}{n+=gc(\$seq);q+=meanqual(\$seq)}END{print \$name,n/NR,q/NR}&#39; &gt;&gt; result.txt&quot;</span> ::: *.fastq.gz

<span class="c1"># 按照 GC 含量排序</span>
$ cat result.txt <span class="p">|</span> sort -n -k3
</pre></div>

<p>R 绘制 gc 分布图，如果 gc 含量偏差超过 10% 时就剔除该基因组数据。</p>
<div class="codehilite"><pre><span></span><span class="o">&gt;</span> <span class="kn">library</span><span class="p">(</span>ggplot2<span class="p">)</span>
<span class="o">&gt;</span> data<span class="o">&lt;-</span>read.table<span class="p">(</span><span class="s">&quot;result.txt&quot;</span><span class="p">)</span>
<span class="o">&gt;</span> qplot<span class="p">(</span>v2<span class="p">,</span> v3<span class="p">,</span> data<span class="o">=</span>data<span class="p">)</span>
</pre></div>

<p>结果显示有一个黑点的GC含量特别低，且Q值特别高。结果显示测序实验 SRR2155551 和 SRR2164197 的 GC 含量仅为22%，而其他样本 GC 含量范围在34%～41%。且这2个实验的样本均为同一个，是1个样本的2此测序（或者数据上传了2次），因此将其剔除。</p>
<h3 id="22">2.2 去除接头<a class="headerlink" href="#22" title="Permanent link">&para;</a></h3>
<p>其次要去除接头的污染。因为分析流程中不仅包括 mapping 的方式，还包含 de novo assembly，为了避免接头序列对基因组拼接的影响，这里最好进行。这里使用的工具是 <code class="codehilite">FastQC</code> 和 <code class="codehilite">fadapa</code></p>
<div class="codehilite"><pre><span></span><span class="c1"># 使用脚本 scan_adaptors.py 来扫描下载的高通量基因组测序数据是否有接头污染的情况。</span>
$ fastqc -t <span class="m">40</span> -q --extract *.fastq.gz
$ python scan_fastqc_report.py -d qc
</pre></div>

<p>接头去除软件很多，但是要自动判断并且批量处理最好自己写脚本比较方便。这里用 python 写一个批量移除的脚本。</p>
<div class="codehilite"><pre><span></span><span class="ch">#!usr/bin/env python</span>
<span class="c1"># -*- coding: utf-8 -*-</span>
<span class="c1"># scripts name: scan_fastqc_report.py</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">Seq</span>
<span class="kn">from</span> <span class="nn">Bio</span> <span class="kn">import</span> <span class="n">SeqIO</span>
<span class="kn">from</span> <span class="nn">Bio.SeqRecord</span> <span class="kn">import</span> <span class="n">SeqRecord</span>
<span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="kn">import</span> <span class="n">generic_nucleotide</span>
<span class="kn">from</span> <span class="nn">fadapa</span> <span class="kn">import</span> <span class="n">Fadapa</span>
<span class="kn">import</span> <span class="nn">argparse</span>

<span class="n">parser</span> <span class="o">=</span> <span class="n">argparse</span><span class="o">.</span><span class="n">ArgumentParser</span><span class="p">()</span>
<span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s1">&#39;-i&#39;</span><span class="p">,</span> <span class="s1">&#39;--input&#39;</span><span class="p">,</span> <span class="n">action</span><span class="o">=</span><span class="s1">&#39;store&#39;</span><span class="p">,</span> <span class="n">dest</span><span class="o">=</span><span class="s1">&#39;qcfile&#39;</span><span class="p">,</span> <span class="nb">type</span><span class="o">=</span><span class="nb">str</span><span class="p">,</span> <span class="n">help</span><span class="o">=</span><span class="s2">&quot;fastqc extract data file&quot;</span><span class="p">)</span>
<span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s1">&#39;-o&#39;</span><span class="p">,</span> <span class="s1">&#39;--output&#39;</span><span class="p">,</span> <span class="n">action</span><span class="o">=</span><span class="s1">&#39;store&#39;</span><span class="p">,</span> <span class="n">dest</span><span class="o">=</span><span class="s1">&#39;adaptor&#39;</span><span class="p">,</span> <span class="nb">type</span><span class="o">=</span><span class="nb">str</span><span class="p">,</span> <span class="n">help</span><span class="o">=</span><span class="s2">&quot;adaptors file&quot;</span><span class="p">)</span>
<span class="n">args</span> <span class="o">=</span> <span class="n">parser</span><span class="o">.</span><span class="n">parse_args</span><span class="p">()</span>
<span class="n">qc_file</span> <span class="o">=</span> <span class="n">Fadapa</span><span class="p">(</span><span class="n">args</span><span class="o">.</span><span class="n">qcfile</span><span class="p">)</span>

<span class="k">if</span> <span class="n">qc_file</span><span class="o">.</span><span class="n">summary</span><span class="p">()[</span><span class="o">-</span><span class="mi">3</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;fail&#39;</span><span class="p">:</span>
    <span class="n">adaptors</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">ors</span> <span class="o">=</span> <span class="n">qc_file</span><span class="o">.</span><span class="n">clean_data</span><span class="p">(</span><span class="s2">&quot;Overrepresented sequences&quot;</span><span class="p">)</span>
    <span class="n">ors</span><span class="o">.</span><span class="n">pop</span><span class="p">(</span><span class="mi">0</span><span class="p">)</span>

    <span class="k">for</span> <span class="p">(</span><span class="n">index</span><span class="p">,</span> <span class="n">seq</span><span class="p">)</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">ors</span><span class="p">):</span>
        <span class="k">if</span> <span class="s1">&#39;NNNNNN&#39;</span> <span class="ow">in</span> <span class="n">seq</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span>
            <span class="k">continue</span>
        <span class="n">adaptors</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">SeqRecord</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="n">seq</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">generic_nucleotide</span><span class="p">),</span> <span class="nb">id</span><span class="o">=</span><span class="s2">&quot;adaptor_</span><span class="si">%d</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">index</span><span class="o">+</span><span class="mi">1</span><span class="p">)))</span>

    <span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">exists</span><span class="p">(</span><span class="n">args</span><span class="o">.</span><span class="n">adaptor</span><span class="p">):</span>
        <span class="n">os</span><span class="o">.</span><span class="n">makedirs</span><span class="p">(</span><span class="n">args</span><span class="o">.</span><span class="n">adaptor</span><span class="p">)</span>
    <span class="n">SeqIO</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">adaptors</span><span class="p">,</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">/adaptor.fasta&quot;</span> <span class="o">%</span> <span class="n">args</span><span class="o">.</span><span class="n">adaptor</span><span class="p">,</span> <span class="s2">&quot;fasta&quot;</span><span class="p">)</span>
    <span class="k">print</span> <span class="s2">&quot;Overrepresented sequences has been save to adaptors.fasta&quot;</span>
<span class="k">else</span><span class="p">:</span>
    <span class="k">print</span> <span class="s2">&quot;No Overrepresented sequences&quot;</span>
</pre></div>

<h2 id="3-pangenome">3. Pangenome 分析<a class="headerlink" href="#3-pangenome" title="Permanent link">&para;</a></h2>
<p>常用的 Pangenome 分析包括基因组组装注释后对基因</p>
<h3 id="31">3.1 基因组组装<a class="headerlink" href="#31" title="Permanent link">&para;</a></h3>
<p>组装细菌基因组最常用的工具是 <code class="codehilite">spades</code>，它可以分批对多个kmer进行组装。</p>
<div class="codehilite"><pre><span></span><span class="c1"># spades 组装</span>
$ <span class="k">for</span> i in <span class="k">$(</span>awk -F<span class="s1">&#39;_&#39;</span> <span class="s1">&#39;!a[$1]++{print}&#39;</span><span class="k">)</span><span class="p">;</span> <span class="k">do</span> spades.py --careful <span class="se">\</span>
&gt; -k <span class="m">21</span>,33,55,77,99,121 -1 <span class="nv">$i_1</span>.fastq.gz -2 <span class="nv">$i_2</span>.fastq.gz -T <span class="m">40</span> -o <span class="nv">$i</span><span class="p">;</span> <span class="k">done</span>
</pre></div>

<p>但是由于下载的批量数据采用不同平台，在不同时间进行的测序，读长彼此差异还是比较大的。比如10K病原基因组计划里，很多样本适用 miseq 进行测序的，读长可达250以上，而很多商业公司或者科研院校则用 hiseq 测序，读长一般在100～150。用同一个参数来组装可能不会获得较好的结果。<code class="codehilite">shovill</code>工具可以帮我们自动化最佳kmer选择，去除低覆盖contigs等操作，一步达到较好的组装序列。</p>
<div class="codehilite"><pre><span></span><span class="c1"># shovill 组装</span>
$ <span class="k">for</span> i in <span class="k">$(</span>awk -F<span class="s1">&#39;_&#39;</span> <span class="s1">&#39;!a[$1]++{print}&#39;</span><span class="k">)</span><span class="p">;</span> <span class="k">do</span> shovill -R1 <span class="nv">$i_1</span>.fastq.gz -R2 <span class="nv">$i_2</span>.fastq.gz -o <span class="nv">$i</span><span class="p">;</span> <span class="k">done</span>
</pre></div>

<h3 id="32">3.2 基因组注释<a class="headerlink" href="#32" title="Permanent link">&para;</a></h3>
<div class="codehilite"><pre><span></span>$ <span class="k">for</span> i in <span class="k">$(</span>awk -F<span class="s1">&#39;_&#39;</span> <span class="s1">&#39;!a[$1]++{print}&#39;</span><span class="k">)</span><span class="p">;</span> <span class="k">do</span> prokka --prefix prokka_<span class="nv">$i</span> <span class="nv">$i</span>/contigs.fa<span class="p">;</span> <span class="k">done</span>
</pre></div>

<h3 id="33-roary">3.3 用 Roary 进行分析<a class="headerlink" href="#33-roary" title="Permanent link">&para;</a></h3>
<div class="codehilite"><pre><span></span>$ cp prokka_*/<span class="nv">$i</span>.gff gff/ <span class="o">&amp;&amp;</span> <span class="nb">cd</span> gff
$ roary -p -e mafft *.gff
</pre></div>

<h3 id="34-harvest">3.4 Harvest<a class="headerlink" href="#34-harvest" title="Permanent link">&para;</a></h3>
<div class="codehilite"><pre><span></span>$ parsnp -g ref.gbk -d data -c -p <span class="m">40</span> -o output
$ harvesttools -i outut/*.ggr -S snp 
</pre></div>

<h2 id="4-snps">4. SNPs 构建进化树<a class="headerlink" href="#4-snps" title="Permanent link">&para;</a></h2>
<div class="codehilite"><pre><span></span>$ raxml -f a -x <span class="m">12345</span> -p <span class="m">12345</span> -#100 -n ban -T <span class="m">40</span> -s snp -m GTRGAMMA
</pre></div>

<h3 id="41-mapping">4.1 基于参考基因组 Mapping 的方法<a class="headerlink" href="#41-mapping" title="Permanent link">&para;</a></h3>
<h4 id="1-snippy">1. Snippy*<a class="headerlink" href="#1-snippy" title="Permanent link">&para;</a></h4>
<div class="codehilite"><pre><span></span><span class="c1"># snippy mapping</span>
$ <span class="k">for</span> i in *.fastq.gz <span class="p">|</span> sort <span class="p">|</span> uniq<span class="p">;</span> <span class="se">\</span>
&gt; <span class="k">do</span> snippy --cpus <span class="m">40</span> --outdir <span class="nv">$i</span> -ref reference.fa <span class="se">\</span>
&gt; --R1 <span class="nv">$i</span>*_1*.fastq.gz --R2 <span class="nv">$i</span>*_2*.fastq.gz<span class="p">;</span> <span class="k">done</span>

<span class="c1"># snippy-core 汇集 snps</span>
$ snippy-core --prefix core-snps SRR* DRR* ERR*

<span class="c1"># raxml 绘制进化树</span>
$ raxml -f a -x <span class="m">12345</span> -p <span class="m">12345</span> -# <span class="m">100</span> -m GTRGAMMA -s core-snps.aln -n ex -T <span class="m">40</span>

<span class="c1"># 将 raxml 进化树结果文件从服务器传到本地电脑，用 figtree 生成图片</span>
$ scp user@server-ip:/path/RAxML_bestTree.ex .
$ figtree RAxML_bestTree.ex <span class="p">&amp;</span>
</pre></div>

<h2 id="5">5. 核心基因分析<a class="headerlink" href="#5" title="Permanent link">&para;</a></h2>
<h3 id="51-get_homologues">5.1 get_homologues<a class="headerlink" href="#51-get_homologues" title="Permanent link">&para;</a></h3>
<h2 id="6">6. 相关软件安装<a class="headerlink" href="#6" title="Permanent link">&para;</a></h2>
<p>*****edirect**</p>
<div class="codehilite"><pre><span></span>$ wget ftp://ftp.ncbi.nlm.nih.gov/entrez/entrezdirect/edirect.tar.gz
$ sudo tar zxf edirect.tar.gz -C /opt
$ sudo chown -R root:root /opt/edirect
$ <span class="nb">cd</span> /opt/edirect <span class="o">&amp;&amp;</span> sudo ./setup.sh
$ sudo ln -s /opt/edirect/* /usr/local/sbin/
</pre></div>

<p><strong>sra-tools</strong></p>
<div class="codehilite"><pre><span></span>$ wget http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/2.7.0/sratoolkit.2.7.0-ubuntu64.tar.gz
$ sudo tar zxf sratoolkit.2.7.0-ubuntu64.tar.gz -C /opt/sratoolkit
$ sudo chown -R root:root /opt/sratoolkit <span class="o">&amp;&amp;</span> <span class="nb">cd</span> /opt/sratoolkit
$ sudo ln -s <span class="sb">`</span><span class="nb">pwd</span><span class="sb">`</span>/bin/* /usr/local/sbin/
</pre></div>

<p><strong>bio-awk</strong></p>
<div class="codehilite"><pre><span></span>$ sudo apt-get install bison
$ git clone https://github.com/lh3/bioawk
$ <span class="nb">cd</span> bioawk <span class="o">&amp;&amp;</span> make
$ sudo cp bioawk /usr/local/sbin
</pre></div>

<p><strong>parallel</strong></p>
<div class="codehilite"><pre><span></span>$ sudo apt install parallel
</pre></div>

<p><strong>gplot</strong></p>
<div class="codehilite"><pre><span></span>$ sudo apt install gnuplot
$ git clone https://github.com/RhysU/gplot
$ sudo cp gplot/gplot /usr/local/sbin
</pre></div>

<p><strong>FastQC</strong></p>
<div class="codehilite"><pre><span></span>$ wget http://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.5.zip
$ sudo unzip fastqc_v0.11.5.zip
$ sudo mv fastqc_v0.11.5 /opt/fastqc <span class="o">&amp;&amp;</span> sudo chown -R root:root /opt/fastqc
$ sudo cp /opt/fastqc/fastqc /usr/local/sbin
</pre></div>

<p><strong>Fadapa</strong></p>
<div class="codehilite"><pre><span></span>$ sudo pip install fadapa
</pre></div>

<p><strong>snippy</strong></p>
<div class="codehilite"><pre><span></span>$ git clone https://github.com/tseemann/snippy
$ sudo cp snippy/bin/* /usr/local/sbin
$ sudo cp snippy/binaries/linux/* /usr/local/sbin
</pre></div>

<p><strong>RAxML</strong></p>
<div class="codehilite"><pre><span></span>$ wget https://github.com/stamatak/standard-RAxML/archive/v8.1.17.tar.gz
$ sudo tar zxf v8.1.17.tar.gz -C /opt/raxml
$ sudo chown -R root:root /opt/raxml <span class="o">&amp;&amp;</span> <span class="nb">cd</span> /opt/raxml
$ sudo make -f Makefile.SSE3.PTHREADS.gcc
$ sudo ln -s <span class="sb">`</span><span class="nb">pwd</span><span class="sb">`</span>/raxmlHPC-PTHREADS-SSE3 /usr/local/raxml
</pre></div>

<h2 id="7-reference">7. Reference<a class="headerlink" href="#7-reference" title="Permanent link">&para;</a></h2>
<ol>
<li>edirect help book: <a href="http://www.ncbi.nlm.nih.gov/books/NBK179288/">http://www.ncbi.nlm.nih.gov/books/NBK179288/</a></li>
<li>sra-tools wiki: <a href="https://github.com/ncbi/sra-tools/wiki">https://github.com/ncbi/sra-tools/wiki</a></li>
<li>parallel tutorial: <a href="https://www.biostars.org/p/63816/">https://www.biostars.org/p/63816/</a></li>
<li>gnuplot tutorial: <a href="http://lzz5235.github.io/2016/01/12/gnuplot.html">http://lzz5235.github.io/2016/01/12/gnuplot.html</a></li>
<li>gplot usage: <a href="https://github.com/RhysU/gplot/blob/master/README.md">https://github.com/RhysU/gplot/blob/master/README.md</a></li>
</ol>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../01_trace-Pae-from-different-source-by-SNP/" title="01. 细菌SNP溯源分析一" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                01. 细菌SNP溯源分析一
              </span>
            </div>
          </a>
        
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2016 - 2018 HZCDC@Lab
          </div>
        
        powered by
        <a href="http://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application.b438e6c5.js"></script>
      
      <script>app.initialize({version:"0.17.1",url:{base:"../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/raphael/2.2.7/raphael.min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/underscore.js/1.8.3/underscore-min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/js-sequence-diagrams/1.0.6/sequence-diagram-min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/flowchart/1.6.5/flowchart.min.js"></script>
      
        <script src="../../assets/js/umlconvert.js"></script>
      
    
    
      
        <script>!function(e,a,t,n,o,c,i){e.GoogleAnalyticsObject=o,e.ga=e.ga||function(){(e.ga.q=e.ga.q||[]).push(arguments)},e.ga.l=1*new Date,c=a.createElement(t),i=a.getElementsByTagName(t)[0],c.async=1,c.src="https://www.google-analytics.com/analytics.js",i.parentNode.insertBefore(c,i)}(window,document,"script",0,"ga"),ga("create","UA-55420348-3","auto"),ga("set","anonymizeIp",!0),ga("send","pageview");var links=document.getElementsByTagName("a");if(Array.prototype.map.call(links,function(e){e.host!=document.location.host&&e.addEventListener("click",function(){var a=e.getAttribute("data-md-action")||"follow";ga("send","event","outbound",a,e.href)})}),document.forms.search){var query=document.forms.search.query;query.addEventListener("blur",function(){if(this.value){var e=document.location.pathname;ga("send","pageview",e+"?q="+this.value)}})}</script>
      
    
  </body>
</html>