



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="病原微生物高通量测序数据分析手册">
      
      
        <link rel="canonical" href="https://indexofire.github.io/pathongs/chapter_10/02_comparative-analysis-Ban-genomes/">
      
      
        <meta name="author" content="Mark Renton">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-0.17.3, mkdocs-material-2.7.3">
    
    
      
        <title>10.2 炭疽芽胞杆菌比较基因组学研究 - Pathogen NGS 数据分析入门</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application.8d40d89b.css">
      
    
    
      <script src="../../assets/javascripts/modernizr.1aa3b519.js"></script>
    
    
      <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
      
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <a href="#snps" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://indexofire.github.io/pathongs" title="Pathogen NGS 数据分析入门" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                Pathogen NGS 数据分析入门
              </span>
              <span class="md-header-nav__topic">
                10.2 炭疽芽胞杆菌比较基因组学研究
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    <span class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </span>
    Pathogen NGS 数据分析入门
  </label>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="前言" class="md-nav__link">
      前言
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2">
    
    <label class="md-nav__link" for="nav-2">
      第一章 数据获取与质控
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        第一章 数据获取与质控
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_01/01_basic-cml/" title="01. Linux 系统操作基本命令" class="md-nav__link">
      01. Linux 系统操作基本命令
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_01/02_linux-installation/" title="02. Linux 安装实例" class="md-nav__link">
      02. Linux 安装实例
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_01/03_linux-administration/" title="03. Linux 系统管理" class="md-nav__link">
      03. Linux 系统管理
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_01/04_awk/" title="04. 数据处理常用工具之 awk" class="md-nav__link">
      04. 数据处理常用工具之 awk
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_01/05_bioawk/" title="‘05. 测序数据工具 bioawk’" class="md-nav__link">
      ‘05. 测序数据工具 bioawk’
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_01/06_python/" title="06. 了解 Python 生活轻松一点" class="md-nav__link">
      06. 了解 Python 生活轻松一点
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_01/07_shell/" title="07. 生信分析要学一点 shell" class="md-nav__link">
      07. 生信分析要学一点 shell
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      第二章 序列处理
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        第二章 序列处理
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_02/01_emboss/" title="’01. EMBOSS 工具集'" class="md-nav__link">
      ’01. EMBOSS 工具集'
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_02/02_blast/" title="02. Blast 与 Blast+" class="md-nav__link">
      02. Blast 与 Blast+
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_02/08_seqkit/" title="03. 国产序列处理神器 SeqKit" class="md-nav__link">
      03. 国产序列处理神器 SeqKit
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_02/09_kmer/" title="04. K-mer 处理工具" class="md-nav__link">
      04. K-mer 处理工具
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      第四章 基因组组装
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        第四章 基因组组装
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_04/03_bacteria-assembly/" title="03. 细菌基因组组装" class="md-nav__link">
      03. 细菌基因组组装
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_04/04_virus-assembly/" title="04. 病毒基因组组装" class="md-nav__link">
      04. 病毒基因组组装
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      第五章 基因组序列注释
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        第五章 基因组序列注释
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_05/01_prokka/" title="01. Prokka 快速注释" class="md-nav__link">
      01. Prokka 快速注释
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_05/02_rast/" title="02. rast" class="md-nav__link">
      02. rast
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../chapter_05/03_PGAAP/" title="03. PGAAP" class="md-nav__link">
      03. PGAAP
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6" checked>
    
    <label class="md-nav__link" for="nav-6">
      第十章 基因组分析实践
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        第十章 基因组分析实践
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../01_trace-Pae-from-different-source-by-SNP/" title="10.1 铜绿假单胞菌不同来源株SNP溯源" class="md-nav__link">
      10.1 铜绿假单胞菌不同来源株SNP溯源
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        10.2 炭疽芽胞杆菌比较基因组学研究
      </label>
    
    <a href="./" title="10.2 炭疽芽胞杆菌比较基因组学研究" class="md-nav__link md-nav__link--active">
      10.2 炭疽芽胞杆菌比较基因组学研究
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1" title="1. 获取公共数据库数据" class="md-nav__link">
    1. 获取公共数据库数据
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2" title="2. 数据的前期处理" class="md-nav__link">
    2. 数据的前期处理
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#21-qc" title="2.1 数据 QC" class="md-nav__link">
    2.1 数据 QC
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#22" title="2.2 去除接头" class="md-nav__link">
    2.2 去除接头
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-pangenome" title="3. Pangenome 分析" class="md-nav__link">
    3. Pangenome 分析
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#31" title="3.1 基因组组装" class="md-nav__link">
    3.1 基因组组装
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#32" title="3.2 基因组注释" class="md-nav__link">
    3.2 基因组注释
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#33-roary" title="3.3 Roary" class="md-nav__link">
    3.3 Roary
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#34-harvest" title="3.4 Harvest" class="md-nav__link">
    3.4 Harvest
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-snps" title="4. SNPs 构建进化树" class="md-nav__link">
    4. SNPs 构建进化树
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#41-mapping" title="4.1 基于参考基因组 Mapping 的方法" class="md-nav__link">
    4.1 基于参考基因组 Mapping 的方法
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5" title="5. 基于组装结果的核心基因组序列" class="md-nav__link">
    5. 基于组装结果的核心基因组序列
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#_1" title="相关软件安装" class="md-nav__link">
    相关软件安装
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reference" title="Reference" class="md-nav__link">
    Reference
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1" title="1. 获取公共数据库数据" class="md-nav__link">
    1. 获取公共数据库数据
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2" title="2. 数据的前期处理" class="md-nav__link">
    2. 数据的前期处理
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#21-qc" title="2.1 数据 QC" class="md-nav__link">
    2.1 数据 QC
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#22" title="2.2 去除接头" class="md-nav__link">
    2.2 去除接头
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-pangenome" title="3. Pangenome 分析" class="md-nav__link">
    3. Pangenome 分析
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#31" title="3.1 基因组组装" class="md-nav__link">
    3.1 基因组组装
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#32" title="3.2 基因组注释" class="md-nav__link">
    3.2 基因组注释
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#33-roary" title="3.3 Roary" class="md-nav__link">
    3.3 Roary
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#34-harvest" title="3.4 Harvest" class="md-nav__link">
    3.4 Harvest
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#4-snps" title="4. SNPs 构建进化树" class="md-nav__link">
    4. SNPs 构建进化树
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#41-mapping" title="4.1 基于参考基因组 Mapping 的方法" class="md-nav__link">
    4.1 基于参考基因组 Mapping 的方法
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#5" title="5. 基于组装结果的核心基因组序列" class="md-nav__link">
    5. 基于组装结果的核心基因组序列
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#_1" title="相关软件安装" class="md-nav__link">
    相关软件安装
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reference" title="Reference" class="md-nav__link">
    Reference
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                
                <h1 id="snps">炭疽芽胞杆菌基因组 SNPs 分析<a class="headerlink" href="#snps" title="Permanent link">&para;</a></h1>
<p>疾控部门属于保障单位，需要加强对生物恐怖事件的防控和应对处置能力。最著名的细菌性生物恐怖战剂就是含有炭疽芽胞杆菌的白色粉末。日常工作中无法接触到这些病原，因此在经验和技能方面较为缺乏。在G20期间，内部也进行了相关项目的应急拉练，除此之外为了提升能力建设，应对可能发生的疑似事件，在实验室快速分析和溯源工作做好能力储备，我们对炭疽芽胞杆菌的基因组数据进行了基本分析，对相关资料和信息进行了储备。</p>
<p>主要通过构建相应的分析流程，获得菌株基因组数据后可以第一时间就进行溯源方面的研究。从而在这准备工作的过程中形成了此文档。主要目的也是为了抛砖引玉，做到举一反三，给其他细菌性病原微生物基因组溯源分析时提供参考。</p>
<p>软件中使用的一些参数是根据自己的服务器设置的，具体运行时需要根据计算机配置进行相应的调整。我们服务器的配置如下：</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<ul>
<li>CPU: E3 2650v3 x2</li>
<li>Memory: ECC 16G x8</li>
<li>Harddisk: 7200转 1T x4 组 Raid 5 + 1 Hotspare</li>
<li>System: Ubuntu 16.04 LTS 64bit</li>
</ul>
</div>
<h2 id="1">1. 获取公共数据库数据<a class="headerlink" href="#1" title="Permanent link">&para;</a></h2>
<p>数据采用 NCBI SRA 数据库中的公共数据来作分析。首先检索 NCBI SRA 数据库中所有使用 illumina 平台对炭疽芽胞杆菌基因组进行 PE 测序的 大于 100M 的 NGS 数据，并在下载这些 sra 数据后转换成 fastq.gz 格式。使用的工具主要为 <code class="codehilite">edirect</code> 和 <code class="codehilite">sra-tools</code>。</p>
<div class="codehilite"><pre><span></span>$ esearch -db sra -query <span class="s1">&#39;&quot;Bacillus anthracis&quot;[Organism] AND \</span>
<span class="s1">&gt; (Hiseq[ALL] OR Miseq[ALL]) AND &quot;strategy wgs&quot;[Properties]&#39;</span> <span class="p">|</span> <span class="se">\</span>
&gt; efetch -db sra -format runinfo <span class="p">|</span> grep <span class="s1">&#39;PAIRED&#39;</span> <span class="p">|</span> grep <span class="s1">&#39;^[DSE]RR*&#39;</span> <span class="p">|</span> <span class="se">\</span>
&gt; awk -F<span class="s1">&#39;,&#39;</span> <span class="s1">&#39;{if($5&gt;100000000) print $1}&#39;</span> <span class="p">|</span> sort -n <span class="p">|</span> <span class="se">\</span>
&gt; xargs fastq-dump --split-files --gzip --outdir bacillus_anthracis
</pre></div>

<p>虽然我们检索时限定了 PE 测序数据，但是很多时候数据库里的信息不准确，还是会有很多 SE 的数据会被下载。确定 PE 或者 SE 的方法很多。</p>
<div class="codehilite"><pre><span></span><span class="c1"># sra-stat 统计 sra 文件，nreads=1 表示 SE 测序，nreads=2 表示 PE 测序。程序统计整个文件，速度比较慢</span>
$ sra-stat -xs *.sra <span class="p">|</span> grep <span class="s2">&quot;nreads&quot;</span>
<span class="c1"># 批量删除</span>
$ parallel <span class="s1">&#39;sra-stat -xs&#39;</span> ::: *.sra <span class="p">|</span> xtract -Pattern Run -element Run@accession <span class="se">\</span>
&gt; Statistics@nreads <span class="p">|</span> awk <span class="s1">&#39;{if($2==1) print $1}&#39;</span> <span class="p">|</span> xargs rm

<span class="c1"># fastq-dump 输出一个 reads，计算行数如果是4，那么就是 SE，如果是8,则是 PE</span>
$ fastq-dump -X <span class="m">1</span> --split-spot -Z SRR955386.sra <span class="p">|</span> wc -l

<span class="c1"># fastq-dump 自己判断，当 sra 文件是 SE 测序时，fastq-dump 只能生成1个 *_1.fastq 文件</span>
$ fastq-dump --split-files ERR493452.sra

<span class="c1"># fastq-dump 只能调用单核转换，可以使用 parallel 工具并行处理加快速度。</span>
$ parallel <span class="s2">&quot;fastq-dump --split-files --gzip --outdir bacillus_anthracis&quot;</span> ::: *.sra
</pre></div>

<p>有时候有些sra文件在 dump 时会出错，不能生成对应的 PE fastq 文件，这就需要比较后删除了。如果是 mate pair 的数据，会生成 <code class="codehilite">*_3.fastq.gz</code>，如果不需要则可手工删除。</p>
<div class="codehilite"><pre><span></span><span class="c1"># 核对 PE 双端测序文件是否一致</span>
$ ls -l *_1.fastq.gz <span class="p">|</span> awk <span class="s1">&#39;{print $9}&#39;</span> <span class="p">|</span> awk -F<span class="s1">&#39;_&#39;</span> <span class="s1">&#39;{print $1}&#39;</span> &gt; R1.txt
$ ls -l *_2.fastq.gz <span class="p">|</span> awk <span class="s1">&#39;{print $9}&#39;</span> <span class="p">|</span> awk -F<span class="s1">&#39;_&#39;</span> <span class="s1">&#39;{print $1}&#39;</span> &gt; R2.txt
$ diff R1.txt R2.txt
<span class="c1"># diff 如果2个文件内容一致，则终端不会有输出。否则就要删除文件</span>
</pre></div>

<h2 id="2">2. 数据的前期处理<a class="headerlink" href="#2" title="Permanent link">&para;</a></h2>
<p>由于是公共数据库下载的数据，并不能保证测序实验质量或者数据提交者是否提交的是质控后的数据。因此建议要对数据做一些前期的质控处理。</p>
<h3 id="21-qc">2.1 数据 QC<a class="headerlink" href="#21-qc" title="Permanent link">&para;</a></h3>
<p>首先对基因组 GC 含量和 Q 值进行初步筛选，对于偏差较大的数据考虑直接剔除（或者用其他软件验证看是否是错误物种）。这里使用的工具为 <code class="codehilite">bioawk</code> 或 <code class="codehilite">parallel</code>。</p>
<div class="codehilite"><pre><span></span><span class="c1"># 因为 reads 覆盖度可能并不均一，高覆盖度的区域 GC 含量比重会略高。但是大部分情况下，正常测序的基因组这种区域相对来说不多，平均到基因组后会整体 GC 含量影响不大。</span>
$ <span class="k">for</span> i in *.fastq.gz<span class="p">;</span> <span class="k">do</span> bioawk -c fastx <span class="s1">&#39;BEGIN{n=0;q=0}{n+=gc($seq);q+=meanqual($seq)}END{print $name,n/NR,q/NR}&#39;</span> <span class="nv">$i</span> &gt;&gt; result.txt<span class="p">;</span> <span class="k">done</span>
<span class="c1"># awk 类工具是单进程的，为了加速可以使用 parallel 来并行计算</span>
$ parallel <span class="s2">&quot;bioawk -c fastx &#39;BEGIN{n=0;q=0}{n+=gc(\$seq);q+=meanqual(\$seq)}END{print \$name,n/NR,q/NR}&#39; &gt;&gt; result.txt&quot;</span> ::: *.fastq.gz
<span class="c1"># 按照 GC 含量排序</span>
$ cat result.txt <span class="p">|</span> sort -n -k3
</pre></div>

<p>R 绘制 gc 分布图，如果 gc 含量偏差超过 10% 时就剔除该基因组数据。</p>
<div class="codehilite"><pre><span></span><span class="o">&gt;</span> <span class="kn">library</span><span class="p">(</span>ggplot2<span class="p">)</span>
<span class="o">&gt;</span> data<span class="o">&lt;-</span>read.table<span class="p">(</span><span class="s">&quot;result.txt&quot;</span><span class="p">)</span>
<span class="o">&gt;</span> qplot<span class="p">(</span>v2<span class="p">,</span> v3<span class="p">,</span> data<span class="o">=</span>data<span class="p">)</span>
</pre></div>

<p>结果显示有一个黑点的GC含量特别低，且Q值特别高。结果显示测序实验 SRR2155551 和 SRR2164197 的 GC 含量仅为22%，而其他样本 GC 含量范围在34%～41%。且这2个实验的样本均为同一个，是1个样本的2此测序（或者数据上传了2次），因此将其剔除。</p>
<h3 id="22">2.2 去除接头<a class="headerlink" href="#22" title="Permanent link">&para;</a></h3>
<p>其次要去除接头的污染。因为分析流程中不仅包括 mapping 的方式，还包含 de novo assembly，为了避免接头序列对基因组拼接的影响，这里最好进行。这里使用的工具是 <code class="codehilite">FastQC</code> 和 <code class="codehilite">fadapa</code></p>
<div class="codehilite"><pre><span></span># 使用脚本 scan_adaptors.py 来扫描下载的高通量基因组测序数据是否有接头污染的情况。
$ fastqc -t 40 -q --extract *.fastq.gz
$ python scan_fastqc_report.py -d qc
</pre></div>

<div class="codehilite"><pre><span></span><span class="ch">#!usr/bin/env python</span>
<span class="c1"># -*- coding: utf-8 -*-</span>
<span class="c1"># scripts name: scan_fastqc_report.py</span>
<span class="kn">import</span> <span class="nn">os</span>
<span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">Seq</span>
<span class="kn">from</span> <span class="nn">Bio</span> <span class="kn">import</span> <span class="n">SeqIO</span>
<span class="kn">from</span> <span class="nn">Bio.SeqRecord</span> <span class="kn">import</span> <span class="n">SeqRecord</span>
<span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="kn">import</span> <span class="n">generic_nucleotide</span>
<span class="kn">from</span> <span class="nn">fadapa</span> <span class="kn">import</span> <span class="n">Fadapa</span>
<span class="kn">import</span> <span class="nn">argparse</span>

<span class="n">parser</span> <span class="o">=</span> <span class="n">argparse</span><span class="o">.</span><span class="n">ArgumentParser</span><span class="p">()</span>
<span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s1">&#39;-i&#39;</span><span class="p">,</span> <span class="s1">&#39;--input&#39;</span><span class="p">,</span> <span class="n">action</span><span class="o">=</span><span class="s1">&#39;store&#39;</span><span class="p">,</span> <span class="n">dest</span><span class="o">=</span><span class="s1">&#39;qcfile&#39;</span><span class="p">,</span> <span class="nb">type</span><span class="o">=</span><span class="nb">str</span><span class="p">,</span> <span class="n">help</span><span class="o">=</span><span class="s2">&quot;fastqc extract data file&quot;</span><span class="p">)</span>
<span class="n">parser</span><span class="o">.</span><span class="n">add_argument</span><span class="p">(</span><span class="s1">&#39;-o&#39;</span><span class="p">,</span> <span class="s1">&#39;--output&#39;</span><span class="p">,</span> <span class="n">action</span><span class="o">=</span><span class="s1">&#39;store&#39;</span><span class="p">,</span> <span class="n">dest</span><span class="o">=</span><span class="s1">&#39;adaptor&#39;</span><span class="p">,</span> <span class="nb">type</span><span class="o">=</span><span class="nb">str</span><span class="p">,</span> <span class="n">help</span><span class="o">=</span><span class="s2">&quot;adaptors file&quot;</span><span class="p">)</span>
<span class="n">args</span> <span class="o">=</span> <span class="n">parser</span><span class="o">.</span><span class="n">parse_args</span><span class="p">()</span>
<span class="n">qc_file</span> <span class="o">=</span> <span class="n">Fadapa</span><span class="p">(</span><span class="n">args</span><span class="o">.</span><span class="n">qcfile</span><span class="p">)</span>

<span class="k">if</span> <span class="n">qc_file</span><span class="o">.</span><span class="n">summary</span><span class="p">()[</span><span class="o">-</span><span class="mi">3</span><span class="p">][</span><span class="mi">0</span><span class="p">]</span> <span class="o">==</span> <span class="s1">&#39;fail&#39;</span><span class="p">:</span>
    <span class="n">adaptors</span> <span class="o">=</span> <span class="p">[]</span>
    <span class="n">ors</span> <span class="o">=</span> <span class="n">qc_file</span><span class="o">.</span><span class="n">clean_data</span><span class="p">(</span><span class="s2">&quot;Overrepresented sequences&quot;</span><span class="p">)</span>
    <span class="n">ors</span><span class="o">.</span><span class="n">pop</span><span class="p">(</span><span class="mi">0</span><span class="p">)</span>

    <span class="k">for</span> <span class="p">(</span><span class="n">index</span><span class="p">,</span> <span class="n">seq</span><span class="p">)</span> <span class="ow">in</span> <span class="nb">enumerate</span><span class="p">(</span><span class="n">ors</span><span class="p">):</span>
        <span class="k">if</span> <span class="s1">&#39;NNNNNN&#39;</span> <span class="ow">in</span> <span class="n">seq</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span>
            <span class="k">continue</span>
        <span class="n">adaptors</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="n">SeqRecord</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="n">seq</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">generic_nucleotide</span><span class="p">),</span> <span class="nb">id</span><span class="o">=</span><span class="s2">&quot;adaptor_</span><span class="si">%d</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">index</span><span class="o">+</span><span class="mi">1</span><span class="p">)))</span>

    <span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">exists</span><span class="p">(</span><span class="n">args</span><span class="o">.</span><span class="n">adaptor</span><span class="p">):</span>
        <span class="n">os</span><span class="o">.</span><span class="n">makedirs</span><span class="p">(</span><span class="n">args</span><span class="o">.</span><span class="n">adaptor</span><span class="p">)</span>
    <span class="n">SeqIO</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">adaptors</span><span class="p">,</span> <span class="s2">&quot;</span><span class="si">%s</span><span class="s2">/adaptor.fasta&quot;</span> <span class="o">%</span> <span class="n">args</span><span class="o">.</span><span class="n">adaptor</span><span class="p">,</span> <span class="s2">&quot;fasta&quot;</span><span class="p">)</span>
    <span class="k">print</span> <span class="s2">&quot;Overrepresented sequences has been save to adaptors.fasta&quot;</span>
<span class="k">else</span><span class="p">:</span>
    <span class="k">print</span> <span class="s2">&quot;No Overrepresented sequences&quot;</span>
</pre></div>

<h2 id="3-pangenome">3. Pangenome 分析<a class="headerlink" href="#3-pangenome" title="Permanent link">&para;</a></h2>
<h3 id="31">3.1 基因组组装<a class="headerlink" href="#31" title="Permanent link">&para;</a></h3>
<div class="codehilite"><pre><span></span>$ spades.py
</pre></div>

<h3 id="32">3.2 基因组注释<a class="headerlink" href="#32" title="Permanent link">&para;</a></h3>
<div class="codehilite"><pre><span></span>$ prokka
</pre></div>

<h3 id="33-roary">3.3 Roary<a class="headerlink" href="#33-roary" title="Permanent link">&para;</a></h3>
<div class="codehilite"><pre><span></span>$ roary -p -e mafft *.gff
</pre></div>

<h3 id="34-harvest">3.4 Harvest<a class="headerlink" href="#34-harvest" title="Permanent link">&para;</a></h3>
<div class="codehilite"><pre><span></span>$
</pre></div>

<h2 id="4-snps">4. SNPs 构建进化树<a class="headerlink" href="#4-snps" title="Permanent link">&para;</a></h2>
<h3 id="41-mapping">4.1 基于参考基因组 Mapping 的方法<a class="headerlink" href="#41-mapping" title="Permanent link">&para;</a></h3>
<p><strong>Snippy</strong></p>
<div class="codehilite"><pre><span></span># snippy mapping
$ for i in *.fastq.gz | sort | uniq; \
&gt; do snippy --cpus 20 --outdir $i -ref reference.fa \
&gt; --R1 $i*_1*.fastq.gz --R2 $i*_2*.fastq.gz; done

# snippy-core 汇集 snps
$ snippy-core --prefix core-snps SRR* DRR* ERR*

# raxml 绘制进化树
$ raxml -f a -x 12345 -p 12345 -# 100 -m GTRGAMMA -s core-snps.aln -n ex -T 40

# 将 raxml 进化树结果文件从服务器传到本地电脑，用 figtree 生成图片
$ scp user@server-ip:/path/RAxML_bestTree.ex .
$ figtree RAxML_bestTree.ex &amp;
</pre></div>

<p><strong>Tools</strong></p>
<h2 id="5">5. 基于组装结果的核心基因组序列<a class="headerlink" href="#5" title="Permanent link">&para;</a></h2>
<h2 id="_1">相关软件安装<a class="headerlink" href="#_1" title="Permanent link">&para;</a></h2>
<p><strong>edirect</strong></p>
<div class="codehilite"><pre><span></span>$ wget ftp://ftp.ncbi.nlm.nih.gov/entrez/entrezdirect/edirect.tar.gz
$ sudo tar zxf edirect.tar.gz -C /opt
$ sudo chown -R root:root /opt/edirect
$ <span class="nb">cd</span> /opt/edirect <span class="o">&amp;&amp;</span> sudo ./setup.sh
$ sudo ln -s /opt/edirect/* /usr/local/sbin/
</pre></div>

<p><strong>sra-tools</strong></p>
<div class="codehilite"><pre><span></span>$ wget http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/2.7.0/sratoolkit.2.7.0-ubuntu64.tar.gz
$ sudo tar zxf sratoolkit.2.7.0-ubuntu64.tar.gz -C /opt/sratoolkit
$ sudo chown -R root:root /opt/sratoolkit <span class="o">&amp;&amp;</span> <span class="nb">cd</span> /opt/sratoolkit
$ sudo ln -s <span class="sb">`</span><span class="nb">pwd</span><span class="sb">`</span>/bin/* /usr/local/sbin/
</pre></div>

<p><strong>bio-awk</strong></p>
<div class="codehilite"><pre><span></span>$ sudo apt-get install bison
$ git clone https://github.com/lh3/bioawk
$ <span class="nb">cd</span> bioawk <span class="o">&amp;&amp;</span> make
$ sudo cp bioawk /usr/local/sbin
</pre></div>

<p><strong>parallel</strong></p>
<div class="codehilite"><pre><span></span>$ sudo apt install parallel
</pre></div>

<p><strong>gplot</strong></p>
<div class="codehilite"><pre><span></span>$ sudo apt install gnuplot
$ git clone https://github.com/RhysU/gplot
$ sudo cp gplot/gplot /usr/local/sbin
</pre></div>

<p><strong>FastQC</strong></p>
<div class="codehilite"><pre><span></span>$ wget http://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.5.zip
$ sudo unzip fastqc_v0.11.5.zip
$ sudo mv fastqc_v0.11.5 /opt/fastqc <span class="o">&amp;&amp;</span> sudo chown -R root:root /opt/fastqc
$ sudo cp /opt/fastqc/fastqc /usr/local/sbin
</pre></div>

<p><strong>Fadapa</strong></p>
<div class="codehilite"><pre><span></span>$ sudo pip install fadapa
</pre></div>

<p><strong>snippy</strong></p>
<div class="codehilite"><pre><span></span>$ git clone https://github.com/tseemann/snippy
$ sudo cp snippy/bin/* /usr/local/sbin
$ sudo cp snippy/binaries/linux/* /usr/local/sbin
</pre></div>

<p><strong>RAxML</strong></p>
<div class="codehilite"><pre><span></span>$ wget https://github.com/stamatak/standard-RAxML/archive/v8.1.17.tar.gz
$ sudo tar zxf v8.1.17.tar.gz -C /opt/raxml
$ sudo chown -R root:root /opt/raxml <span class="o">&amp;&amp;</span> <span class="nb">cd</span> /opt/raxml
$ sudo make -f Makefile.SSE3.PTHREADS.gcc
$ sudo ln -s <span class="sb">`</span><span class="nb">pwd</span><span class="sb">`</span>/raxmlHPC-PTHREADS-SSE3 /usr/local/raxml
</pre></div>

<h2 id="reference">Reference<a class="headerlink" href="#reference" title="Permanent link">&para;</a></h2>
<ol>
<li>edirect help book: <a href="http://www.ncbi.nlm.nih.gov/books/NBK179288/">http://www.ncbi.nlm.nih.gov/books/NBK179288/</a></li>
<li>sra-tools wiki: <a href="https://github.com/ncbi/sra-tools/wiki">https://github.com/ncbi/sra-tools/wiki</a></li>
<li>parallel tutorial: <a href="https://www.biostars.org/p/63816/">https://www.biostars.org/p/63816/</a></li>
<li>gnuplot tutorial: <a href="http://lzz5235.github.io/2016/01/12/gnuplot.html">http://lzz5235.github.io/2016/01/12/gnuplot.html</a></li>
<li>gplot usage: <a href="https://github.com/RhysU/gplot/blob/master/README.md">https://github.com/RhysU/gplot/blob/master/README.md</a></li>
</ol>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../01_trace-Pae-from-different-source-by-SNP/" title="10.1 铜绿假单胞菌不同来源株SNP溯源" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                10.1 铜绿假单胞菌不同来源株SNP溯源
              </span>
            </div>
          </a>
        
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2016 - 2018 HZCDC@Lab
          </div>
        
        powered by
        <a href="http://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application.b438e6c5.js"></script>
      
      <script>app.initialize({version:"0.17.3",url:{base:"../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/raphael/2.2.7/raphael.min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/underscore.js/1.8.3/underscore-min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/js-sequence-diagrams/1.0.6/sequence-diagram-min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/flowchart/1.6.5/flowchart.min.js"></script>
      
        <script src="../../assets/js/umlconvert.js"></script>
      
    
    
      
    
  </body>
</html>