



<!DOCTYPE html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      <meta http-equiv="x-ua-compatible" content="ie=edge">
      
        <meta name="description" content="病原微生物高通量测序数据分析手册">
      
      
        <link rel="canonical" href="https://indexofire.github.io/pathongs/C08_Genome-Visualization/03_circos/">
      
      
        <meta name="author" content="Mark Renton">
      
      
        <meta name="lang:clipboard.copy" content="Copy to clipboard">
      
        <meta name="lang:clipboard.copied" content="Copied to clipboard">
      
        <meta name="lang:search.language" content="en">
      
        <meta name="lang:search.pipeline.stopwords" content="True">
      
        <meta name="lang:search.pipeline.trimmer" content="True">
      
        <meta name="lang:search.result.none" content="No matching documents">
      
        <meta name="lang:search.result.one" content="1 matching document">
      
        <meta name="lang:search.result.other" content="# matching documents">
      
        <meta name="lang:search.tokenizer" content="[\s\-]+">
      
      <link rel="shortcut icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-0.17.1, mkdocs-material-2.7.3">
    
    
      
        <title>03. circos 工具使用 - Pathogen NGS 数据分析入门</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/application.8d40d89b.css">
      
    
    
      <script src="../../assets/javascripts/modernizr.1aa3b519.js"></script>
    
    
      <link href="https://fonts.gstatic.com" rel="preconnect" crossorigin>
      
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700|Roboto+Mono">
        <style>body,input{font-family:"Roboto","Helvetica Neue",Helvetica,Arial,sans-serif}code,kbd,pre{font-family:"Roboto Mono","Courier New",Courier,monospace}</style>
      
      <link rel="stylesheet" href="https://fonts.googleapis.com/icon?family=Material+Icons">
    
    
    
  </head>
  
    <body dir="ltr">
  
    <svg class="md-svg">
      <defs>
        
        
          <svg xmlns="http://www.w3.org/2000/svg" width="416" height="448"
    viewBox="0 0 416 448" id="github">
  <path fill="currentColor" d="M160 304q0 10-3.125 20.5t-10.75 19-18.125
        8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19 18.125-8.5
        18.125 8.5 10.75 19 3.125 20.5zM320 304q0 10-3.125 20.5t-10.75
        19-18.125 8.5-18.125-8.5-10.75-19-3.125-20.5 3.125-20.5 10.75-19
        18.125-8.5 18.125 8.5 10.75 19 3.125 20.5zM360
        304q0-30-17.25-51t-46.75-21q-10.25 0-48.75 5.25-17.75 2.75-39.25
        2.75t-39.25-2.75q-38-5.25-48.75-5.25-29.5 0-46.75 21t-17.25 51q0 22 8
        38.375t20.25 25.75 30.5 15 35 7.375 37.25 1.75h42q20.5 0
        37.25-1.75t35-7.375 30.5-15 20.25-25.75 8-38.375zM416 260q0 51.75-15.25
        82.75-9.5 19.25-26.375 33.25t-35.25 21.5-42.5 11.875-42.875 5.5-41.75
        1.125q-19.5 0-35.5-0.75t-36.875-3.125-38.125-7.5-34.25-12.875-30.25-20.25-21.5-28.75q-15.5-30.75-15.5-82.75
        0-59.25 34-99-6.75-20.5-6.75-42.5 0-29 12.75-54.5 27 0 47.5 9.875t47.25
        30.875q36.75-8.75 77.25-8.75 37 0 70 8 26.25-20.5
        46.75-30.25t47.25-9.75q12.75 25.5 12.75 54.5 0 21.75-6.75 42 34 40 34
        99.5z" />
</svg>
        
      </defs>
    </svg>
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="search" autocomplete="off">
    <label class="md-overlay" data-md-component="overlay" for="drawer"></label>
    
      <a href="#circos" tabindex="1" class="md-skip">
        Skip to content
      </a>
    
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header-nav md-grid">
    <div class="md-flex">
      <div class="md-flex__cell md-flex__cell--shrink">
        <a href="https://indexofire.github.io/pathongs" title="Pathogen NGS 数据分析入门" class="md-header-nav__button md-logo">
          
            <i class="md-icon"></i>
          
        </a>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        <label class="md-icon md-icon--menu md-header-nav__button" for="drawer"></label>
      </div>
      <div class="md-flex__cell md-flex__cell--stretch">
        <div class="md-flex__ellipsis md-header-nav__title" data-md-component="title">
          
            
              <span class="md-header-nav__topic">
                Pathogen NGS 数据分析入门
              </span>
              <span class="md-header-nav__topic">
                03. circos 工具使用
              </span>
            
          
        </div>
      </div>
      <div class="md-flex__cell md-flex__cell--shrink">
        
          
            <label class="md-icon md-icon--search md-header-nav__button" for="search"></label>
            
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="query" data-md-state="active">
      <label class="md-icon md-search__icon" for="search"></label>
      <button type="reset" class="md-icon md-search__icon" data-md-component="reset" tabindex="-1">
        &#xE5CD;
      </button>
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="result">
          <div class="md-search-result__meta">
            Type to start searching
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
          
        
      </div>
      
        <div class="md-flex__cell md-flex__cell--shrink">
          <div class="md-header-nav__source">
            


  


  <a href="https://github.com/indexofire/pathongs/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      GitHub
    </div>
  </a>

          </div>
        </div>
      
    </div>
  </nav>
</header>
    
    <div class="md-container">
      
        
      
      
      <main class="md-main">
        <div class="md-main__inner md-grid" data-md-component="container">
          
            
              <div class="md-sidebar md-sidebar--primary" data-md-component="navigation">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    <nav class="md-nav md-nav--primary" data-md-level="0">
  <label class="md-nav__title md-nav__title--site" for="drawer">
    <span class="md-nav__button md-logo">
      
        <i class="md-icon"></i>
      
    </span>
    Pathogen NGS 数据分析入门
  </label>
  
    <div class="md-nav__source">
      


  


  <a href="https://github.com/indexofire/pathongs/" title="Go to repository" class="md-source" data-md-source="github">
    
      <div class="md-source__icon">
        <svg viewBox="0 0 24 24" width="24" height="24">
          <use xlink:href="#github" width="24" height="24"></use>
        </svg>
      </div>
    
    <div class="md-source__repository">
      GitHub
    </div>
  </a>

    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      


  <li class="md-nav__item">
    <a href="../.." title="第零章 开始之前" class="md-nav__link">
      第零章 开始之前
    </a>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-2" type="checkbox" id="nav-2">
    
    <label class="md-nav__link" for="nav-2">
      第一章 背景知识介绍
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-2">
        第一章 背景知识介绍
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/01_linux-installation/" title="01. Linux 安装实例" class="md-nav__link">
      01. Linux 安装实例
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/02_basic-cml/" title="02. Linux 基本命令" class="md-nav__link">
      02. Linux 基本命令
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/03_linux-software/" title="03. Linux 软件管理" class="md-nav__link">
      03. Linux 软件管理
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/04_awk/" title="04. 列处理工具 awk" class="md-nav__link">
      04. 列处理工具 awk
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/05_bioawk/" title="05. awk 的扩展 bioawk" class="md-nav__link">
      05. awk 的扩展 bioawk
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/06_python/" title="06. Python 编程" class="md-nav__link">
      06. Python 编程
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/07_shell/" title="07. Shell脚本" class="md-nav__link">
      07. Shell脚本
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/08_perl/" title="08. Perl 编程" class="md-nav__link">
      08. Perl 编程
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C01_Linux-and-Script-Language/09_r/" title="09. R 编程" class="md-nav__link">
      09. R 编程
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-3" type="checkbox" id="nav-3">
    
    <label class="md-nav__link" for="nav-3">
      第二章 产用工具
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-3">
        第二章 产用工具
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C02_Common-Utility/01_databases/" title="01. 公共生物数据库" class="md-nav__link">
      01. 公共生物数据库
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C02_Common-Utility/02_blast/" title="02. Blast 与 Blast+" class="md-nav__link">
      02. Blast 与 Blast+
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C02_Common-Utility/03_edirect/" title="03. 玩转 edirect" class="md-nav__link">
      03. 玩转 edirect
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C02_Common-Utility/04_emboss/" title="04. EMBOSS 工具集" class="md-nav__link">
      04. EMBOSS 工具集
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C02_Common-Utility/07_mummer/" title="07. MUMmer 工具" class="md-nav__link">
      07. MUMmer 工具
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C02_Common-Utility/08_seqkit/" title="08. SeqKit 工具" class="md-nav__link">
      08. SeqKit 工具
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C02_Common-Utility/09_kmer/" title="09. Kmer 工具" class="md-nav__link">
      09. Kmer 工具
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-4" type="checkbox" id="nav-4">
    
    <label class="md-nav__link" for="nav-4">
      第三章 各类数据文件格式
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-4">
        第三章 各类数据文件格式
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C03_File-Format/01_fasta/" title="01. fasta" class="md-nav__link">
      01. fasta
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C03_File-Format/02_embl/" title="02. embl" class="md-nav__link">
      02. embl
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C03_File-Format/03_plain-txt/" title="03. plain text" class="md-nav__link">
      03. plain text
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-5" type="checkbox" id="nav-5">
    
    <label class="md-nav__link" for="nav-5">
      第四章 基因组组装
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-5">
        第四章 基因组组装
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C04_Pathogen-Genome-Assembly/01_bwa-samtools-pipeline/" title="01. bwa samtools 流程" class="md-nav__link">
      01. bwa samtools 流程
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C04_Pathogen-Genome-Assembly/02_gatk/" title="02. GATK 流程" class="md-nav__link">
      02. GATK 流程
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C04_Pathogen-Genome-Assembly/03_bacteria-assembly/" title="03. 细菌基因组组装" class="md-nav__link">
      03. 细菌基因组组装
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C04_Pathogen-Genome-Assembly/04_virus-assembly/" title="04. 病毒基因组组装" class="md-nav__link">
      04. 病毒基因组组装
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C04_Pathogen-Genome-Assembly/05_quast/" title="05. Quast 验证" class="md-nav__link">
      05. Quast 验证
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-6" type="checkbox" id="nav-6">
    
    <label class="md-nav__link" for="nav-6">
      第五章 基因组比对
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-6">
        第五章 基因组比对
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C05_Pathogen-Mapping-Pipeline/01_bwa-long-sequence-mapping/" title="01. bwa 长序列比对" class="md-nav__link">
      01. bwa 长序列比对
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-7" type="checkbox" id="nav-7">
    
    <label class="md-nav__link" for="nav-7">
      第六章 基因组注释
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-7">
        第六章 基因组注释
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C06_Genome-Annotation/01_prokka/" title="01. Prokka 快速注释" class="md-nav__link">
      01. Prokka 快速注释
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C06_Genome-Annotation/02_rast/" title="02. 在线注释工具 rast" class="md-nav__link">
      02. 在线注释工具 rast
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C06_Genome-Annotation/03_PGAAP/" title="03. 注释工具 PGAAP" class="md-nav__link">
      03. 注释工具 PGAAP
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C06_Genome-Annotation/04_snpEFF/" title="04. SNP注释工具 snpEFF" class="md-nav__link">
      04. SNP注释工具 snpEFF
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-8" type="checkbox" id="nav-8">
    
    <label class="md-nav__link" for="nav-8">
      第七章 Pan和Core基因组分析
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-8">
        第七章 Pan和Core基因组分析
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C07_PanGenome-and-CoreGenome/01_pangenome/" title="01. Pangenome 概述" class="md-nav__link">
      01. Pangenome 概述
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C07_PanGenome-and-CoreGenome/02_get_homologues/" title="02. Get_Homologues 分析" class="md-nav__link">
      02. Get_Homologues 分析
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C07_PanGenome-and-CoreGenome/03_BPGA/" title="03. BPGA 分析" class="md-nav__link">
      03. BPGA 分析
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C07_PanGenome-and-CoreGenome/04_roary/" title="04. Roary 分析" class="md-nav__link">
      04. Roary 分析
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C07_PanGenome-and-CoreGenome/05_panseq/" title="05. panseq 分析" class="md-nav__link">
      05. panseq 分析
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C07_PanGenome-and-CoreGenome/06_gubbins/" title="06. Gubbins 分析" class="md-nav__link">
      06. Gubbins 分析
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C07_PanGenome-and-CoreGenome/07_orthmcl/" title="07. OrthMCL 分析" class="md-nav__link">
      07. OrthMCL 分析
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C07_PanGenome-and-CoreGenome/08_ITEP/" title="08. ITEP 分析" class="md-nav__link">
      08. ITEP 分析
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      

  


  <li class="md-nav__item md-nav__item--active md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-9" type="checkbox" id="nav-9" checked>
    
    <label class="md-nav__link" for="nav-9">
      第八章 基因组可视化
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-9">
        第八章 基因组可视化
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../01_igv/" title="01. igv 查看比对结果" class="md-nav__link">
      01. igv 查看比对结果
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../02_BRIG/" title="02. BRIG 分析工具" class="md-nav__link">
      02. BRIG 分析工具
    </a>
  </li>

        
          
          
          

  


  <li class="md-nav__item md-nav__item--active">
    
    <input class="md-toggle md-nav__toggle" data-md-toggle="toc" type="checkbox" id="toc">
    
      
    
    
      <label class="md-nav__link md-nav__link--active" for="toc">
        03. circos 工具使用
      </label>
    
    <a href="./" title="03. circos 工具使用" class="md-nav__link md-nav__link--active">
      03. circos 工具使用
    </a>
    
      
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1" title="1. 安装" class="md-nav__link">
    1. 安装
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2" title="2. 输入文件类型" class="md-nav__link">
    2. 输入文件类型
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#21" title="2.1 数据文件" class="md-nav__link">
    2.1 数据文件
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#22" title="2.2 配置文件" class="md-nav__link">
    2.2 配置文件
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#221-circosconf" title="2.2.1 circos.conf" class="md-nav__link">
    2.2.1 circos.conf
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#222-ideogramconf" title="2.2.2 ideogram.conf" class="md-nav__link">
    2.2.2 ideogram.conf
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#223-imageconf" title="2.2.3 image.conf" class="md-nav__link">
    2.2.3 image.conf
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#224-highlightconf" title="2.2.4 highlight.conf" class="md-nav__link">
    2.2.4 highlight.conf
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#225-ticksconf" title="2.2.5 ticks.conf" class="md-nav__link">
    2.2.5 ticks.conf
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3" title="3. 可视化的组成要素" class="md-nav__link">
    3. 可视化的组成要素
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-circos" title="3. 用circos来绘图" class="md-nav__link">
    3. 用circos来绘图
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#31" title="3.1 配置文件" class="md-nav__link">
    3.1 配置文件
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#32" title="3.2 绘制测序拼接结果信息" class="md-nav__link">
    3.2 绘制测序拼接结果信息
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#321" title="3.2.1 数据准备" class="md-nav__link">
    3.2.1 数据准备
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#322" title="3.2.2 配置文件" class="md-nav__link">
    3.2.2 配置文件
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#33" title="3.3. 绘制一个比较基因组学数据" class="md-nav__link">
    3.3. 绘制一个比较基因组学数据
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#331" title="3.3.1 数据准备" class="md-nav__link">
    3.3.1 数据准备
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#332" title="3.3.2 配置文件" class="md-nav__link">
    3.3.2 配置文件
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#circos_1" title="解析 circos" class="md-nav__link">
    解析 circos
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#circos_2" title="circos 的衍生产品" class="md-nav__link">
    circos 的衍生产品
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reference" title="Reference" class="md-nav__link">
    Reference
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
    
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-10" type="checkbox" id="nav-10">
    
    <label class="md-nav__link" for="nav-10">
      第九章 分析平台
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-10">
        第九章 分析平台
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C09_Analysis-Platform/01_docker/" title="01. 基于 docker 镜像" class="md-nav__link">
      01. 基于 docker 镜像
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C09_Analysis-Platform/02_galaxy/" title="02. Galaxy 平台" class="md-nav__link">
      02. Galaxy 平台
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C09_Analysis-Platform/03_amazon/" title="03. Amazon EC2 云平台" class="md-nav__link">
      03. Amazon EC2 云平台
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C09_Analysis-Platform/04_others/" title="04. 其他平台" class="md-nav__link">
      04. 其他平台
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-11" type="checkbox" id="nav-11">
    
    <label class="md-nav__link" for="nav-11">
      第十章 重组与种群分析
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-11">
        第十章 重组与种群分析
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C10_Recombination-and-Population-Structure/01_virus-recombination/" title="01. 病毒重组" class="md-nav__link">
      01. 病毒重组
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-12" type="checkbox" id="nav-12">
    
    <label class="md-nav__link" for="nav-12">
      第十一章 常见分析工具
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-12">
        第十一章 常见分析工具
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C11_Pathogen-NGS-Application/01_srst2/" title="01. SRST2 短序列比对基因工具" class="md-nav__link">
      01. SRST2 短序列比对基因工具
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C11_Pathogen-NGS-Application/02_mlst/" title="02. 拼接序列管家基因扫描工具 MLST" class="md-nav__link">
      02. 拼接序列管家基因扫描工具 MLST
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C11_Pathogen-NGS-Application/03_ariba/" title="03. 拼接序列耐药基因扫描工具 Ariba" class="md-nav__link">
      03. 拼接序列耐药基因扫描工具 Ariba
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C11_Pathogen-NGS-Application/04_csibelia/" title="04. 细菌共线性分析工具 C-Sibelia" class="md-nav__link">
      04. 细菌共线性分析工具 C-Sibelia
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-13" type="checkbox" id="nav-13">
    
    <label class="md-nav__link" for="nav-13">
      第十二章 宏基因组分析
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-13">
        第十二章 宏基因组分析
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/01_kraken/" title="01. kraken" class="md-nav__link">
      01. kraken
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/02_centrifuge/" title="02. centrifuge" class="md-nav__link">
      02. centrifuge
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/03_bracken/" title="03. bracken" class="md-nav__link">
      03. bracken
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/04_sourmash/" title="04. sourmash" class="md-nav__link">
      04. sourmash
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/05_clark/" title="05. clark" class="md-nav__link">
      05. clark
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/06_krona/" title="06. krona" class="md-nav__link">
      06. krona
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/07_pavian/" title="07. pavian" class="md-nav__link">
      07. pavian
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/08_supri/" title="08. supri" class="md-nav__link">
      08. supri
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/09_shotgun/" title="‘09. shotgun 分析’" class="md-nav__link">
      ‘09. shotgun 分析’
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C12_Metagenomics-Analysis/10_16s/" title="10. 16S 种群分析" class="md-nav__link">
      10. 16S 种群分析
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-14" type="checkbox" id="nav-14">
    
    <label class="md-nav__link" for="nav-14">
      第十三章 单分子测序
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-14">
        第十三章 单分子测序
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C13_Single-Molecular-Sequencing/01_nanopore/" title="01. nanopore" class="md-nav__link">
      01. nanopore
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C13_Single-Molecular-Sequencing/02_pacbio/" title="02. pacbio" class="md-nav__link">
      02. pacbio
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C13_Single-Molecular-Sequencing/03_canu/" title="03. canu" class="md-nav__link">
      03. canu
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C13_Single-Molecular-Sequencing/04_trimming/" title="04. trimming" class="md-nav__link">
      04. trimming
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C13_Single-Molecular-Sequencing/06_hybrid/" title="06. hybrid" class="md-nav__link">
      06. hybrid
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-15" type="checkbox" id="nav-15">
    
    <label class="md-nav__link" for="nav-15">
      第十四章 进化树构建
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-15">
        第十四章 进化树构建
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C14_Phylogeny-Analysis/01_msa/" title="01. 多重序列比对" class="md-nav__link">
      01. 多重序列比对
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C14_Phylogeny-Analysis/02_raxml/" title="02. RAxML" class="md-nav__link">
      02. RAxML
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-16" type="checkbox" id="nav-16">
    
    <label class="md-nav__link" for="nav-16">
      第十五章 基因组流行病学
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-16">
        第十五章 基因组流行病学
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C15_Pathogen-Geospatial-Analysis/04_cartodb/" title="04. CartoDB" class="md-nav__link">
      04. CartoDB
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
      
      
      


  <li class="md-nav__item md-nav__item--nested">
    
      <input class="md-toggle md-nav__toggle" data-md-toggle="nav-17" type="checkbox" id="nav-17">
    
    <label class="md-nav__link" for="nav-17">
      第十六章 基因组分析实践
    </label>
    <nav class="md-nav" data-md-component="collapsible" data-md-level="1">
      <label class="md-nav__title" for="nav-17">
        第十六章 基因组分析实践
      </label>
      <ul class="md-nav__list" data-md-scrollfix>
        
        
          
          
          


  <li class="md-nav__item">
    <a href="../../C16_Analysis-Examples/01_trace-Pae-from-different-source-by-SNP/" title="01. 细菌SNP溯源分析一" class="md-nav__link">
      01. 细菌SNP溯源分析一
    </a>
  </li>

        
          
          
          


  <li class="md-nav__item">
    <a href="../../C16_Analysis-Examples/02_comparative-analysis-Ban-genomes/" title="02. 细菌比较基因组学研究" class="md-nav__link">
      02. 细菌比较基因组学研究
    </a>
  </li>

        
      </ul>
    </nav>
  </li>

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              <div class="md-sidebar md-sidebar--secondary" data-md-component="toc">
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary">
  
  
    
  
  
    <label class="md-nav__title" for="toc">Table of contents</label>
    <ul class="md-nav__list" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#1" title="1. 安装" class="md-nav__link">
    1. 安装
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#2" title="2. 输入文件类型" class="md-nav__link">
    2. 输入文件类型
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#21" title="2.1 数据文件" class="md-nav__link">
    2.1 数据文件
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#22" title="2.2 配置文件" class="md-nav__link">
    2.2 配置文件
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#221-circosconf" title="2.2.1 circos.conf" class="md-nav__link">
    2.2.1 circos.conf
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#222-ideogramconf" title="2.2.2 ideogram.conf" class="md-nav__link">
    2.2.2 ideogram.conf
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#223-imageconf" title="2.2.3 image.conf" class="md-nav__link">
    2.2.3 image.conf
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#224-highlightconf" title="2.2.4 highlight.conf" class="md-nav__link">
    2.2.4 highlight.conf
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#225-ticksconf" title="2.2.5 ticks.conf" class="md-nav__link">
    2.2.5 ticks.conf
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3" title="3. 可视化的组成要素" class="md-nav__link">
    3. 可视化的组成要素
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#3-circos" title="3. 用circos来绘图" class="md-nav__link">
    3. 用circos来绘图
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#31" title="3.1 配置文件" class="md-nav__link">
    3.1 配置文件
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#32" title="3.2 绘制测序拼接结果信息" class="md-nav__link">
    3.2 绘制测序拼接结果信息
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#321" title="3.2.1 数据准备" class="md-nav__link">
    3.2.1 数据准备
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#322" title="3.2.2 配置文件" class="md-nav__link">
    3.2.2 配置文件
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#33" title="3.3. 绘制一个比较基因组学数据" class="md-nav__link">
    3.3. 绘制一个比较基因组学数据
  </a>
  
    <nav class="md-nav">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#331" title="3.3.1 数据准备" class="md-nav__link">
    3.3.1 数据准备
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#332" title="3.3.2 配置文件" class="md-nav__link">
    3.3.2 配置文件
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#circos_1" title="解析 circos" class="md-nav__link">
    解析 circos
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#circos_2" title="circos 的衍生产品" class="md-nav__link">
    circos 的衍生产品
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#reference" title="Reference" class="md-nav__link">
    Reference
  </a>
  
</li>
      
      
      
      
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/indexofire/pathongs/edit/master/docs/C08_Genome-Visualization/03_circos.md" title="Edit this page" class="md-icon md-content__icon">&#xE3C9;</a>
                
                
                <h1 id="circos"><a href="http://circos.ca/">circos</a><a class="headerlink" href="#circos" title="Permanent link">&para;</a></h1>
<p><a href="http://circos.ca/">circos</a> 是一个关系可视化圈图表达软件，常被用来绘制环状的细菌染色体或质粒，或作为可视化比较基因组学研究多个细菌基因组之间差异的工具。<a href="http://circos.ca/">circos</a> 的优点是生成的图形优美，可以表示各种类型的数据。但是缺点在于牵涉内容过于繁杂，采用的术语名称的含义往往让人摸不着头脑。教程也不够明确。<a href="http://circos.ca/">circos</a> 的目的是绘制图形，有点类似Photoshop，它提供的是工具，而不是一步速成的获得某种图形的方法。</p>
<h2 id="1">1. 安装<a class="headerlink" href="#1" title="Permanent link">&para;</a></h2>
<p><strong>下载安装</strong></p>
<div class="codehilite"><pre><span></span><span class="c1"># 安装依赖包</span>
<span class="c1"># install gd2-xpm development library</span>
$ sudo apt-get install libgd2-xpm-dev

<span class="c1"># link circos perl import path. circos use &quot;#!/usr/env perl&quot; instead of</span>
<span class="c1"># &quot;/usr/bin/env&quot; which is default in ubuntu</span>
$ sudo ln -s /usr/bin/env /bin/env

<span class="c1"># install perl modules needed in circos</span>
$ sudo cpan install Clone <span class="se">\</span>
&gt; Config::General <span class="se">\</span>
&gt; Font::TTF <span class="se">\</span>
&gt; List::MoreUtils <span class="se">\</span>
&gt; Math::Bezier <span class="se">\</span>
&gt; Math::VecStat <span class="se">\</span>
&gt; Math::Round <span class="se">\</span>
&gt; Params::Validate <span class="se">\</span>
&gt; Readonly <span class="se">\</span>
&gt; Regexp::Common <span class="se">\</span>
&gt; Set::IntSpan <span class="se">\</span>
&gt; Text::Format <span class="se">\</span>
&gt; GD <span class="se">\</span>
&gt; Statistics::Basic

<span class="c1"># 安装 circos</span>
<span class="c1"># 从官方下载安装包</span>
$ wet http://circos.ca/distribution/circos-0.69.tgz
$ tar xfz circos-0.69.tgz -C ~/apps/circos

<span class="c1"># 添加cirocs到系统路径中</span>
$ sudo ln -s ~/apps/circos/bin/circos /usr/local/sbin/circos

<span class="c1"># 或者只添加到当前用户环境中</span>
$ <span class="nb">echo</span> <span class="s2">&quot;export PATH=\$PATH:\$HOME/apps/circos/bin/&quot;</span> &gt;&gt; ~/.bashrc
$ <span class="nb">source</span> ~/.bashrc
$ circos -v

<span class="c1"># 如果屏幕打印出以下信息，说明你可以正常使用circos了。</span>
circos <span class="p">|</span> v <span class="m">0</span>.69 <span class="p">|</span> <span class="m">6</span> Dec <span class="m">2015</span> <span class="p">|</span> Perl <span class="m">5</span>.018002
</pre></div>

<p>如果 cpan 下载很慢或者无法连接，可以修改 mirror 来改善。如果你的 CPAN 是安装在个人用户环境里，将 <code class="codehilite">~/.cpan/CPAN/MyConfig.pm</code> 文件里的 urllist 行后面的内容修改成离自己最近的镜像站点，比如：</p>
<div class="codehilite"><pre><span></span>&#39;urllist&#39; =&gt; [q[http://mirrors.sohu.com/CPAN/]],
</pre></div>

<p><strong>使用 conda 安装</strong></p>
<div class="codehilite"><pre><span></span><span class="c1"># 创建一个虚拟环境安装 circos</span>
$ conda create env circos
$ <span class="nb">source</span> activate circos
<span class="o">(</span>circos<span class="o">)</span>$ conda install circos
</pre></div>

<h2 id="2">2. 输入文件类型<a class="headerlink" href="#2" title="Permanent link">&para;</a></h2>
<p><a href="http://circos.ca/">circos</a> 主要需要2类文件，一类是可视化结果的数据文件，一类是用于告诉 <a href="http://circos.ca/">circos</a> 如何展示的配置文件。</p>
<h3 id="21">2.1 数据文件<a class="headerlink" href="#21" title="Permanent link">&para;</a></h3>
<p><a href="http://circos.ca/">circos</a> 只负责画图，图中要展示的数据需要自己预先准备好。比如基因组的GC含量，需要用其他程序先生成以下这种类型的文本格式文件。</p>
<div class="codehilite"><pre><span></span>染色体 开始位置 结束位置 颜色值 其他选项[可选]
chr1  1000    2000  red
</pre></div>

<h3 id="22">2.2 配置文件<a class="headerlink" href="#22" title="Permanent link">&para;</a></h3>
<p>一般绘图需要以下配置文件：</p>
<ul>
<li>circos.conf</li>
<li>ideogram.conf</li>
<li>ticks.conf</li>
<li>image.conf</li>
<li>highlight.conf</li>
<li>colors_fonts_patterns.conf</li>
</ul>
<p>你可以将配置文件内容全部放在circos.conf里，也可以分离成单独的文件通过include方式导入。</p>
<h4 id="221-circosconf">2.2.1 circos.conf<a class="headerlink" href="#221-circosconf" title="Permanent link">&para;</a></h4>
<p>这是circos主配置文件，其他的配置文件可以通过include的方式来导入。</p>
<div class="codehilite"><pre><span></span>&lt;&lt;include your_other_configure.conf&gt;&gt;
</pre></div>

<h4 id="222-ideogramconf">2.2.2 ideogram.conf<a class="headerlink" href="#222-ideogramconf" title="Permanent link">&para;</a></h4>
<h4 id="223-imageconf">2.2.3 image.conf<a class="headerlink" href="#223-imageconf" title="Permanent link">&para;</a></h4>
<h4 id="224-highlightconf">2.2.4 highlight.conf<a class="headerlink" href="#224-highlightconf" title="Permanent link">&para;</a></h4>
<h4 id="225-ticksconf">2.2.5 ticks.conf<a class="headerlink" href="#225-ticksconf" title="Permanent link">&para;</a></h4>
<h2 id="3">3. 可视化的组成要素<a class="headerlink" href="#3" title="Permanent link">&para;</a></h2>
<p>\<plots\>\<\/plots>，\<links\>\<\/links> 区块内的参数是属于 Global 的。
\<plot\>\<\/plot> 区块内的参数是属于 Local 的。</p>
<p>如果是某个参数在多个block中都重复使用，那么可以设置成 global 的参数，当某个block要使用新的值时，再用 local 的参数覆盖即可。</p>
<div class="codehilite"><pre><span></span># plot可以理解为图层，多个圈图图层构成一个plots
<span class="nt">&lt;plot&gt;&lt;/plot&gt;</span>

# plots 由
# type 描述图层内容类型，如text
type = text

# r0, r1 表示圈图半径，<span class="nt">&lt;1在内圈</span><span class="err">，</span><span class="nt">&gt;</span>1在外圈, r0-r1为宽度
r0 = 0.95r
r1 = 0.85r
</pre></div>

<ol>
<li>ideogram</li>
<li>highlight</li>
<li>tick</li>
<li>label</li>
<li>link</li>
</ol>
<p>circos.conf 参数</p>
<div class="codehilite"><pre><span></span># 染色体的数据集
karyotype = data.txt

# 定义圈图标尺
chromosomes_units = 1000

# 如果设置成yes，所有ideogram上都会现实ticks
chromosomes_display_default = yes
</pre></div>

<h2 id="3-circos">3. 用circos来绘图<a class="headerlink" href="#3-circos" title="Permanent link">&para;</a></h2>
<p>绘制环状 Salmonella LT2 基因组</p>
<p>之前的教程里介绍了用DNAPlotter绘制一个基因组完成图的圈图，圆圈内容strand，GC content和GC skew。从某个角度上看，DNAplotter的配置文件与circos有一定的类似，2个软件的区别在于前者专注于绘制单个细菌基因组，后者可以完成更多的功能。第一个例子我们通过用circos来绘制类似DNAplotter生成的基因组圈图，从而学习circos的基本用法，</p>
<p>准备数据文件：从NCBI上下载Salonella LT基因组数据文件。</p>
<div class="codehilite"><pre><span></span>$ esearch -db nuccore -query <span class="s2">&quot;NC_003197.1[accn]&quot;</span> <span class="p">|</span> efetch -db nuccore -format fasta &gt; LT2.fasta
$ esearch -db nuccore -query <span class="s2">&quot;NC_003197.1[accn]&quot;</span> <span class="p">|</span> efetch -db nuccore -format gb &gt; LT2.gb
</pre></div>

<p>绘制circos所需要的ideogram所需要的几类数据，我们通过一个python脚本get_circos_data.py来获得。</p>
<div class="codehilite"><pre><span></span><span class="c1"># -*- utf-8 -*-</span>
<span class="kn">from</span> <span class="nn">Bio</span> <span class="kn">import</span> <span class="n">SeqIO</span>

<span class="n">rec</span> <span class="o">=</span> <span class="n">SeqIO</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="s2">&quot;LT2.gb&quot;</span><span class="p">,</span> <span class="s2">&quot;genbank&quot;</span><span class="p">)</span>
<span class="n">forward</span> <span class="o">=</span> <span class="p">[]</span>
<span class="n">reverse</span> <span class="o">=</span> <span class="p">[]</span>

<span class="k">for</span> <span class="n">feature</span> <span class="ow">in</span> <span class="n">rec</span><span class="o">.</span><span class="n">features</span><span class="p">:</span>
    <span class="k">if</span> <span class="n">feature</span><span class="o">.</span><span class="n">type</span> <span class="o">!=</span> <span class="s1">&#39;gene&#39;</span><span class="p">:</span>
        <span class="k">continue</span>
    <span class="k">if</span> <span class="n">feature</span><span class="o">.</span><span class="n">strand</span> <span class="o">==</span> <span class="mi">1</span><span class="p">:</span>
        <span class="n">forward</span><span class="o">.</span><span class="n">append</span><span class="p">((</span><span class="nb">int</span><span class="p">(</span><span class="n">feature</span><span class="o">.</span><span class="n">location</span><span class="o">.</span><span class="n">start</span><span class="p">),</span> <span class="nb">int</span><span class="p">(</span><span class="n">feature</span><span class="o">.</span><span class="n">location</span><span class="o">.</span><span class="n">end</span><span class="p">)))</span>
    <span class="k">elif</span> <span class="n">feature</span><span class="o">.</span><span class="n">strand</span> <span class="o">==</span> <span class="o">-</span><span class="mi">1</span><span class="p">:</span>
        <span class="n">reverse</span><span class="o">.</span><span class="n">append</span><span class="p">((</span><span class="nb">int</span><span class="p">(</span><span class="n">feature</span><span class="o">.</span><span class="n">location</span><span class="o">.</span><span class="n">start</span><span class="p">),</span> <span class="nb">int</span><span class="p">(</span><span class="n">feature</span><span class="o">.</span><span class="n">location</span><span class="o">.</span><span class="n">end</span><span class="p">)))</span>

<span class="k">def</span> <span class="nf">get_forward_strand</span><span class="p">(</span><span class="n">strands</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Create forward strands</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s2">&quot;forward.txt&quot;</span><span class="p">,</span> <span class="s2">&quot;w&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
        <span class="k">for</span> <span class="n">strand</span> <span class="ow">in</span> <span class="n">strands</span><span class="p">:</span>
            <span class="n">f</span><span class="o">.</span><span class="n">writelines</span><span class="p">(</span><span class="s2">&quot;chr1</span><span class="se">\t</span><span class="si">%d</span><span class="se">\t</span><span class="si">%d</span><span class="se">\t</span><span class="s2">fillcolor=chr4</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">strand</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">strand</span><span class="p">[</span><span class="mi">1</span><span class="p">]))</span>

<span class="k">def</span> <span class="nf">get_reverse_strand</span><span class="p">(</span><span class="n">strands</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Create forward strands</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s2">&quot;reverse.txt&quot;</span><span class="p">,</span> <span class="s2">&quot;w&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
        <span class="k">for</span> <span class="n">strand</span> <span class="ow">in</span> <span class="n">strands</span><span class="p">:</span>
            <span class="n">f</span><span class="o">.</span><span class="n">writelines</span><span class="p">(</span><span class="s2">&quot;chr1</span><span class="se">\t</span><span class="si">%d</span><span class="se">\t</span><span class="si">%d</span><span class="se">\t</span><span class="s2">fillcolor=chr6</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">strand</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span> <span class="n">strand</span><span class="p">[</span><span class="mi">1</span><span class="p">]))</span>

<span class="k">def</span> <span class="nf">calc_GC_content</span><span class="p">(</span><span class="n">seq</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Return GC content of input sequence</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">gc</span> <span class="o">=</span> <span class="nb">sum</span><span class="p">(</span><span class="n">seq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="n">x</span><span class="p">)</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="p">[</span><span class="s1">&#39;G&#39;</span><span class="p">,</span><span class="s1">&#39;C&#39;</span><span class="p">,</span><span class="s1">&#39;g&#39;</span><span class="p">,</span><span class="s1">&#39;c&#39;</span><span class="p">])</span>
    <span class="n">gc_content</span> <span class="o">=</span> <span class="n">gc</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="nb">len</span><span class="p">(</span><span class="n">seq</span><span class="p">))</span>
    <span class="k">return</span> <span class="nb">round</span><span class="p">(</span><span class="n">gc_content</span><span class="p">,</span> <span class="mi">4</span><span class="p">)</span>

<span class="k">def</span> <span class="nf">calc_GC_skew</span><span class="p">(</span><span class="n">seq</span><span class="p">):</span>
    <span class="sd">&quot;&quot;&quot;</span>
<span class="sd">    Reuturn GC skew of input sequence</span>
<span class="sd">    &quot;&quot;&quot;</span>
    <span class="n">g</span> <span class="o">=</span> <span class="n">seq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s1">&#39;G&#39;</span><span class="p">)</span> <span class="o">+</span> <span class="n">seq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s1">&#39;g&#39;</span><span class="p">)</span>
    <span class="n">c</span> <span class="o">=</span> <span class="n">seq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s1">&#39;C&#39;</span><span class="p">)</span> <span class="o">+</span> <span class="n">seq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s1">&#39;c&#39;</span><span class="p">)</span>
    <span class="k">try</span><span class="p">:</span>
        <span class="n">skew</span> <span class="o">=</span> <span class="p">(</span><span class="n">g</span> <span class="o">-</span> <span class="n">c</span><span class="p">)</span><span class="o">/</span><span class="nb">float</span><span class="p">(</span><span class="n">g</span> <span class="o">+</span> <span class="n">c</span><span class="p">)</span>
    <span class="k">except</span> <span class="ne">ZeroDivisionError</span><span class="p">:</span>
        <span class="n">skew</span> <span class="o">=</span> <span class="mi">0</span>

    <span class="k">return</span> <span class="nb">round</span><span class="p">(</span><span class="n">skew</span><span class="p">,</span> <span class="mi">4</span><span class="p">)</span>

<span class="n">window</span> <span class="o">=</span> <span class="mi">10000</span>
<span class="n">step</span> <span class="o">=</span> <span class="mi">5000</span>
<span class="n">seqs</span> <span class="o">=</span> <span class="n">SeqIO</span><span class="o">.</span><span class="n">parse</span><span class="p">(</span><span class="s2">&quot;LT2.fasta&quot;</span><span class="p">,</span> <span class="s2">&quot;fasta&quot;</span><span class="p">)</span>

<span class="k">for</span> <span class="n">seq</span> <span class="ow">in</span> <span class="n">seqs</span><span class="p">:</span>
    <span class="n">seq_1</span> <span class="o">=</span> <span class="n">seq</span><span class="o">.</span><span class="n">seq</span>

<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s2">&quot;gc_skew.txt&quot;</span><span class="p">,</span> <span class="s2">&quot;w&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f1</span><span class="p">:</span>
    <span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s2">&quot;gc_content.txt&quot;</span><span class="p">,</span> <span class="s2">&quot;w&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f2</span><span class="p">:</span>
        <span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">seq_1</span><span class="p">),</span> <span class="n">step</span><span class="p">):</span>
            <span class="n">subseq</span> <span class="o">=</span> <span class="n">seq_1</span><span class="p">[</span><span class="n">i</span><span class="p">:</span><span class="n">i</span><span class="o">+</span><span class="n">window</span><span class="p">]</span>
            <span class="n">gc_content</span> <span class="o">=</span> <span class="p">(</span><span class="n">calc_GC_content</span><span class="p">(</span><span class="n">subseq</span><span class="p">))</span>
            <span class="n">gc_skew</span> <span class="o">=</span> <span class="p">(</span><span class="n">calc_GC_skew</span><span class="p">(</span><span class="n">subseq</span><span class="p">))</span>
            <span class="n">start</span> <span class="o">=</span> <span class="p">(</span><span class="n">i</span> <span class="o">+</span> <span class="mi">1</span> <span class="k">if</span> <span class="p">(</span><span class="n">i</span><span class="o">+</span><span class="mi">1</span><span class="o">&lt;=</span><span class="nb">len</span><span class="p">(</span><span class="n">seq_1</span><span class="p">))</span> <span class="k">else</span> <span class="n">i</span><span class="p">)</span>
            <span class="n">end</span> <span class="o">=</span> <span class="p">(</span> <span class="n">i</span> <span class="o">+</span> <span class="n">step</span> <span class="k">if</span> <span class="p">(</span><span class="n">i</span><span class="o">+</span> <span class="n">step</span><span class="o">&lt;=</span><span class="nb">len</span><span class="p">(</span><span class="n">seq_1</span><span class="p">))</span> <span class="k">else</span> <span class="nb">len</span><span class="p">(</span><span class="n">seq_1</span><span class="p">))</span>
            <span class="n">f1</span><span class="o">.</span><span class="n">writelines</span><span class="p">(</span><span class="s2">&quot;chr1</span><span class="se">\t</span><span class="si">%d</span><span class="se">\t</span><span class="si">%d</span><span class="se">\t</span><span class="si">%s</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">start</span><span class="p">,</span> <span class="n">end</span><span class="p">,</span> <span class="n">gc_skew</span><span class="p">))</span>
            <span class="n">f2</span><span class="o">.</span><span class="n">writelines</span><span class="p">(</span><span class="s2">&quot;chr1</span><span class="se">\t</span><span class="si">%d</span><span class="se">\t</span><span class="si">%d</span><span class="se">\t</span><span class="si">%s</span><span class="se">\n</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">start</span><span class="p">,</span> <span class="n">end</span><span class="p">,</span> <span class="n">gc_content</span><span class="p">))</span>

<span class="n">get_forward_strand</span><span class="p">(</span><span class="n">forward</span><span class="p">)</span>
<span class="n">get_reverse_strand</span><span class="p">(</span><span class="n">reverse</span><span class="p">)</span>
</pre></div>

<p>运行脚本：</p>
<div class="codehilite"><pre><span></span>$ python get_circos_data.py
</pre></div>

<h3 id="31">3.1 配置文件<a class="headerlink" href="#31" title="Permanent link">&para;</a></h3>
<div class="codehilite"><pre><span></span>$ <span class="nb">cd</span> circos_data
$ touch circos.conf
</pre></div>

<p>大部分细菌的基因组都是单染色体的，因此一个细菌基因完成图可以用1个完整的圈来表示。通过设置spacing来去除间断。</p>
<div class="codehilite"><pre><span></span><span class="nt">&lt;spacing&gt;</span>
  default = u
  break = u
<span class="nt">&lt;/spacing&gt;</span>
</pre></div>

<p>circos.conf 文件添加以下内容</p>
<div class="codehilite"><pre><span></span><span class="err">&lt;</span><span class="nt">&lt;include</span> <span class="err">colors_fonts_patterns.conf</span><span class="nt">&gt;</span>&gt;
<span class="nt">&lt;ideogram&gt;</span>
  <span class="nt">&lt;spacing&gt;</span>
    default  = 0u
    break    = 0u
  <span class="nt">&lt;/spacing&gt;</span>
  radius    = 0.85r
  thickness  = 5p
  stroke_thickness  = 1
  stroke_color = black
  fill = yes
  fill_color = black
<span class="nt">&lt;/ideogram&gt;</span>

<span class="nt">&lt;image&gt;</span>
dir  = .
file  = circos.png
png  = yes
svg  = yes

# radius of inscribed circle in image
radius        = 1000p

# by default angle=0 is at 3 o&#39;clock position
angle_offset      = -90

#angle_orientation = counterclockwise

auto_alpha_colors = yes
auto_alpha_steps  = 5
background = white
<span class="nt">&lt;/image&gt;</span>

karyotype = data/node.txt
chromosome_units = 1000
chromosomes_display_default = yes
show_ticks          = yes
show_tick_labels    = yes
show_grid          = no
grid_start          = dims(ideogram,radius_inner) - 0.5r
grid_end            = dims(ideogram,radius_inner)

# --------------------- ticks start --------------------- #

<span class="nt">&lt;ticks&gt;</span>
# if yes, skip of label of 0bp
skip_first_label = no
# if yes, skip of lable of last
skip_last_label = no

# ticks radius to show, outside or inside
radius  = dims(ideogram,radius_outer)

# genome size * label_multiplier = label units
label_multiplier = 1e-6

tick_separation = 2p
min_label_distance_to_edge = 0p
label_separation = 5p
# label distance to ticks
label_offset    = 5p
label_size = 8p
size = 20p

<span class="nt">&lt;tick&gt;</span>
spacing        = 10000u
color          = gray
show_label      = no
thickness      = 1p
<span class="nt">&lt;/tick&gt;</span>

<span class="nt">&lt;tick&gt;</span>
spacing        = 100000u
color          = black
show_label      = yes
label_size      = 16p
thickness      = 2p

# label format, %.1f = 1.1, %d = 1, %.2f = 1.10
format          = %.1f
grid            = yes
grid_color      = white
grid_thickness  = 1p
<span class="nt">&lt;/tick&gt;</span>

<span class="nt">&lt;tick&gt;</span>
spacing        = 1000000u
size            = 30p
color          = black
show_label      = yes
label_size      = 24p
thickness      = 3p
# label format, %.1f = 1.1, %d = 1, %.2f = 1.10
format          = %d
# plus Mb to lables
suffix          = &quot; Mb&quot;
grid            = yes
grid_color      = dgrey
grid_thickness  = 1p
<span class="nt">&lt;/tick&gt;</span>
<span class="nt">&lt;/ticks&gt;</span>
# --------------------- ticks ends ---------------------- #

# --------------------- plot start ---------------------- #
<span class="nt">&lt;plots&gt;</span>
  <span class="nt">&lt;plot&gt;</span>
    type  = tile
    file  = data/forward.txt
    r1    = 0.95r
    r0    = 0.90r
    margin = 1b
    stroke_thickness = 0
    color = orange
    orientation = center
    # tile height
    thickness = 20p
    # different layer index height
    padding = 5p
    layers = 1
    <span class="nt">&lt;rules&gt;</span>
      <span class="nt">&lt;rule&gt;</span>
        condition = var(size) &gt; 5kb
        color = black
      <span class="nt">&lt;/rule&gt;</span>
    <span class="nt">&lt;/rules&gt;</span>
  <span class="nt">&lt;/plot&gt;</span>

  <span class="nt">&lt;plot&gt;</span>
    type  = tile
    file  = data/reverse.txt
    r1    = 0.92r
    r0    = 0.87r
    margin = 1b
    stroke_thickness = 0
    color = blue
    thickness = 20p
    padding = 5p
    # display one layer of plot
    layers = 1
    <span class="nt">&lt;rule&gt;</span>
      condition = var(size) &gt; 3kb
      color = black
    <span class="nt">&lt;/rule&gt;</span>
  <span class="nt">&lt;/plot&gt;</span>

  <span class="nt">&lt;plot&gt;</span>
    type = line
    r1 = 0.85r
    r0 = 0.75r
    file = data/gc_skew.txt
    thickness = 1
    max_gap = 1u
    color  = vdgrey
    # max/min value of line plotting
    min    = -0.15
    max    = 0.15
    fill_color = vdgrey_a3
    <span class="nt">&lt;backgrounds&gt;</span>
      <span class="nt">&lt;background&gt;</span>
        color    = vvlgreen
        y0        = 0.05
      <span class="nt">&lt;/background&gt;</span>
      <span class="nt">&lt;background&gt;</span>
        color    = vvlred
        y1        = -0.05
      <span class="nt">&lt;/background&gt;</span>
    <span class="nt">&lt;/backgrounds&gt;</span>

    <span class="nt">&lt;axes&gt;</span>
      <span class="nt">&lt;axis&gt;</span>
        color    = lgrey_a2
        thickness = 1
        spacing  = 0.04r
      <span class="nt">&lt;/axis&gt;</span>
    <span class="nt">&lt;/axes&gt;</span>
  <span class="nt">&lt;/plot&gt;</span>

  <span class="nt">&lt;plot&gt;</span>
    type = histogram
    r1 = 0.70r
    r0 = 0.60r
    file = data/gc_content.txt
    thickness = 1
    max_gap = 1u
    min    = 0
    max    = 1
    <span class="nt">&lt;rules&gt;</span>
      <span class="nt">&lt;rule&gt;</span>
        condition = 1
        fill_color = eval(sprintf(&quot;spectral-9-div-%d&quot;, remap_int(var(value), 0,1,1,9)))
      <span class="nt">&lt;/rule&gt;</span>
    <span class="nt">&lt;/rules&gt;</span>
  <span class="nt">&lt;/plot&gt;</span>
<span class="nt">&lt;/plots&gt;</span>

<span class="err">&lt;</span><span class="nt">&lt;include</span> <span class="err">housekeeping.conf</span><span class="nt">&gt;</span>&gt;
</pre></div>

<h3 id="32">3.2 绘制测序拼接结果信息<a class="headerlink" href="#32" title="Permanent link">&para;</a></h3>
<h4 id="321">3.2.1 数据准备<a class="headerlink" href="#321" title="Permanent link">&para;</a></h4>
<h4 id="322">3.2.2 配置文件<a class="headerlink" href="#322" title="Permanent link">&para;</a></h4>
<h3 id="33">3.3. 绘制一个比较基因组学数据<a class="headerlink" href="#33" title="Permanent link">&para;</a></h3>
<p>BRIG 常被用来绘制的多个细菌比较基因组学数据。本例用 circos 绘制类似的圈图，通过结果可视化来发现和展示生物学信息。</p>
<h4 id="331">3.3.1 数据准备<a class="headerlink" href="#331" title="Permanent link">&para;</a></h4>
<h4 id="332">3.3.2 配置文件<a class="headerlink" href="#332" title="Permanent link">&para;</a></h4>
<h2 id="circos_1">解析 circos<a class="headerlink" href="#circos_1" title="Permanent link">&para;</a></h2>
<p>以 <a href="http://jura.wi.mit.edu/bio/education/hot_topics/Circos/Circos.pdf">mit</a> 的circos图为例，可以看出一个circos绘制的圈图中可以包含什么类型的图示。</p>
<p><img alt="circos picture" src="../../assets/images/6.2/circos-anatomy.png" /></p>
<ul>
<li>histogram 柱状图</li>
<li>ideogram 表意文字</li>
<li>heatmap 热力图</li>
<li>links 关系线条</li>
<li>highlights 高亮强调区域</li>
<li>grids 网格图</li>
<li>ticks 圈图标尺</li>
</ul>
<hr />
<div class="admonition note">
<p class="admonition-title">完整circos配置文件结构"</p>
</div>
<div class="codehilite"><pre><span></span># ---- 定义图的颜色信息 ---- #
<span class="nt">&lt;colors&gt;</span>
<span class="nt">&lt;/colors&gt;</span>

# ---- 定义图的字体信息 ---- #
<span class="nt">&lt;fonts&gt;</span>
<span class="nt">&lt;/fonts&gt;</span>

# ---- 定义填充模式图 ---- #
<span class="nt">&lt;patterns&gt;</span>
<span class="nt">&lt;/patterns&gt;</span>

# ---- 图像的基本设置信息 ---- #
<span class="nt">&lt;image&gt;</span>
<span class="nt">&lt;/image&gt;</span>

# ---- ideogram 图 ---- #
<span class="nt">&lt;ideogram&gt;</span>
  <span class="nt">&lt;spacing&gt;</span>
    ideograms之间的间隔
    <span class="nt">&lt;pairwise&gt;</span>
      ideograms对（如人染色体）之间的距离
    <span class="nt">&lt;/pairwise&gt;</span>
  <span class="nt">&lt;/spacing&gt;</span>
<span class="nt">&lt;/ideogram&gt;</span>

# ---- 标尺信息 ---- #
<span class="nt">&lt;ticks&gt;</span>
  <span class="nt">&lt;tick&gt;</span>
    各个tick的配置
  <span class="nt">&lt;/tick&gt;</span>
<span class="nt">&lt;/ticks&gt;</span>

# ---- 区域的放大缩小信息 ---- #
<span class="nt">&lt;zooms&gt;</span>
  <span class="nt">&lt;zoom&gt;</span>
  <span class="nt">&lt;/zoom&gt;</span>
<span class="nt">&lt;/zooms&gt;</span>

# ---- highlights 数据 ---- #
<span class="nt">&lt;highlights&gt;</span>
  <span class="nt">&lt;highlight&gt;</span>
    <span class="nt">&lt;rules&gt;</span>
      <span class="nt">&lt;rule&gt;</span>
        某一个具体 data track 的配置修改
      <span class="nt">&lt;/rule&gt;</span>
    <span class="nt">&lt;/rules&gt;</span>
  <span class="nt">&lt;/highlight&gt;</span>
<span class="nt">&lt;/highlights&gt;</span>

# ---- Data tracks 数据 ---- #
<span class="nt">&lt;plots&gt;</span>
  <span class="nt">&lt;plot&gt;</span>
    <span class="nt">&lt;rules&gt;</span>
      <span class="nt">&lt;rule&gt;</span>
        某一个具体 data track 的配置修改
      <span class="nt">&lt;/rule&gt;</span>
    <span class="nt">&lt;/rules&gt;</span>
    <span class="nt">&lt;axes&gt;</span>
      <span class="nt">&lt;axis&gt;</span>
        plot 的半径轴
      <span class="nt">&lt;/axis&gt;</span>
    <span class="nt">&lt;/axes&gt;</span>
    <span class="nt">&lt;backgrounds&gt;</span>
      <span class="nt">&lt;background&gt;</span>
        plot 的背景设置
      <span class="nt">&lt;/background&gt;</span>
    <span class="nt">&lt;/backgrounds&gt;</span>
  <span class="nt">&lt;/plot&gt;</span>
<span class="nt">&lt;/plots&gt;</span>

# ---- links 数据 ---- #
<span class="nt">&lt;links&gt;</span>
  <span class="nt">&lt;link&gt;</span>
    <span class="nt">&lt;rules&gt;</span>
      <span class="nt">&lt;rule&gt;</span>
        某一个具体 link 数据的配置修改
      <span class="nt">&lt;/rule&gt;</span>
    <span class="nt">&lt;/rules&gt;</span>
  <span class="nt">&lt;/link&gt;</span>
<span class="nt">&lt;/links&gt;</span>
</pre></div>

<div class="admonition note">
<p class="admonition-title">circos配置的单位概念</p>
</div>
<p>一共有4种单位：p, r, u, b</p>
<ul>
<li>p表示像素，1p表示1像素</li>
<li>r表示相对大小，0.95r表示95% ring 大小。</li>
<li>u表示相对chromosomes_unit的长度，如果chromosomes_unit = 1000，则1u就是千分之一的染色体长度。</li>
<li>b表示碱基，如果染色体长1M，那么1b就是百万分之一的长度。</li>
</ul>
<h2 id="circos_2"><a href="http://circos.ca/">circos</a> 的衍生产品<a class="headerlink" href="#circos_2" title="Permanent link">&para;</a></h2>
<h2 id="reference">Reference<a class="headerlink" href="#reference" title="Permanent link">&para;</a></h2>
<ul>
<li><a href="http://circos.ca/">Circos Official Website</a></li>
<li><a href="http://jura.wi.mit.edu/bio/education/hot_topics/Circos/Circos.pdf">Creating Circos Plots</a></li>
<li><a href="https://github.com/abremges/2015-pseudo">Plot of the complete Pseudomonas pseudoalcaligenes CECT5344 genome</a></li>
</ul>
<p><code class="codehilite">&quot;` &quot;```````` )&quot;&quot;&quot;&quot;&quot;&quot;&quot;&quot;```````````````</code></p>
                
                  
                
              
              
                


              
            </article>
          </div>
        </div>
      </main>
      
        
<footer class="md-footer">
  
    <div class="md-footer-nav">
      <nav class="md-footer-nav__inner md-grid">
        
          <a href="../02_BRIG/" title="02. BRIG 分析工具" class="md-flex md-footer-nav__link md-footer-nav__link--prev" rel="prev">
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-back md-footer-nav__button"></i>
            </div>
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Previous
                </span>
                02. BRIG 分析工具
              </span>
            </div>
          </a>
        
        
          <a href="../../C09_Analysis-Platform/01_docker/" title="01. 基于 docker 镜像" class="md-flex md-footer-nav__link md-footer-nav__link--next" rel="next">
            <div class="md-flex__cell md-flex__cell--stretch md-footer-nav__title">
              <span class="md-flex__ellipsis">
                <span class="md-footer-nav__direction">
                  Next
                </span>
                01. 基于 docker 镜像
              </span>
            </div>
            <div class="md-flex__cell md-flex__cell--shrink">
              <i class="md-icon md-icon--arrow-forward md-footer-nav__button"></i>
            </div>
          </a>
        
      </nav>
    </div>
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
          <div class="md-footer-copyright__highlight">
            Copyright &copy; 2016 - 2018 HZCDC@Lab
          </div>
        
        powered by
        <a href="http://www.mkdocs.org">MkDocs</a>
        and
        <a href="https://squidfunk.github.io/mkdocs-material/">
          Material for MkDocs</a>
      </div>
      
        
      
    </div>
  </div>
</footer>
      
    </div>
    
      <script src="../../assets/javascripts/application.b438e6c5.js"></script>
      
      <script>app.initialize({version:"0.17.1",url:{base:"../.."}})</script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/raphael/2.2.7/raphael.min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/underscore.js/1.8.3/underscore-min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/js-sequence-diagrams/1.0.6/sequence-diagram-min.js"></script>
      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/flowchart/1.6.5/flowchart.min.js"></script>
      
        <script src="../../assets/js/umlconvert.js"></script>
      
    
    
      
        <script>!function(e,a,t,n,o,c,i){e.GoogleAnalyticsObject=o,e.ga=e.ga||function(){(e.ga.q=e.ga.q||[]).push(arguments)},e.ga.l=1*new Date,c=a.createElement(t),i=a.getElementsByTagName(t)[0],c.async=1,c.src="https://www.google-analytics.com/analytics.js",i.parentNode.insertBefore(c,i)}(window,document,"script",0,"ga"),ga("create","UA-55420348-3","auto"),ga("set","anonymizeIp",!0),ga("send","pageview");var links=document.getElementsByTagName("a");if(Array.prototype.map.call(links,function(e){e.host!=document.location.host&&e.addEventListener("click",function(){var a=e.getAttribute("data-md-action")||"follow";ga("send","event","outbound",a,e.href)})}),document.forms.search){var query=document.forms.search.query;query.addEventListener("blur",function(){if(this.value){var e=document.location.pathname;ga("send","pageview",e+"?q="+this.value)}})}</script>
      
    
  </body>
</html>